<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Mucosal transcriptomic landscape along the small and large intestines in individuals with and without type 2 diabetes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Data_analysis_to_github_files/libs/clipboard/clipboard.min.js"></script>
<script src="Data_analysis_to_github_files/libs/quarto-html/quarto.js"></script>
<script src="Data_analysis_to_github_files/libs/quarto-html/popper.min.js"></script>
<script src="Data_analysis_to_github_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Data_analysis_to_github_files/libs/quarto-html/anchor.min.js"></script>
<link href="Data_analysis_to_github_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Data_analysis_to_github_files/libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Data_analysis_to_github_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Data_analysis_to_github_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Data_analysis_to_github_files/libs/bootstrap/bootstrap-973236bd072d72a04ee9cd82dcc9cb29.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Mucosal transcriptomic landscape along the small and large intestines in individuals with and without type 2 diabetes</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="packages" class="level1">
<h1>Packages</h1>
<section id="colors-and-functions" class="level2">
<h2 class="anchored" data-anchor-id="colors-and-functions">Colors and functions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>getPalette <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">"Spectral"</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>PCA_all <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#4292c6"</span>, <span class="st">"#41ab5d"</span>, <span class="st">"#9e9ac8"</span>,  <span class="st">"#fd8d3c"</span>, <span class="st">"#df65b0"</span>,<span class="st">"#9ecae1"</span>, <span class="st">"#c7e9c0"</span> , <span class="st">"#dadaeb"</span>, <span class="st">"#fdd0a2"</span>, <span class="st">"#f768a1"</span>,<span class="st">"#4292c6"</span>, <span class="st">"#41ab5d"</span>, <span class="st">"#9e9ac8"</span>,  <span class="st">"#fd8d3c"</span>, <span class="st">"#df65b0"</span>,<span class="st">"#9ecae1"</span>) </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>PCA_dim23 <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"#b2182b"</span>, <span class="st">"#d6604d"</span>, <span class="st">"#f4a582"</span>, <span class="st">"#fddbc7"</span>, <span class="st">"#f7f7f7"</span>,  <span class="co"># Red → Yellow → Green (1-5)</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"#d1e5f0"</span>, <span class="st">"#92c5de"</span>, <span class="st">"#4393c3"</span>, <span class="st">"#2166ac"</span>, <span class="st">"#253494"</span>,  <span class="co"># Green variations (6-10)</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"#c51b7d"</span>, <span class="st">"#e9a3c9"</span>, <span class="st">"#fde0ef"</span>, <span class="st">"#e6f5d0"</span>, <span class="st">"#a1d76a"</span>, <span class="st">"#4d9221"</span>)  <span class="co"># Blue → Purple (11-16)</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>PCA_small <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#4292c6"</span>, <span class="st">"#41ab5d"</span>, <span class="st">"#9e9ac8"</span>,  <span class="st">"#fd8d3c"</span>, <span class="st">"#df65b0"</span>,<span class="st">"#9ecae1"</span>, <span class="st">"#c7e9c0"</span> , <span class="st">"#dadaeb"</span>, <span class="st">"#fdd0a2"</span>, <span class="st">"#f768a1"</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>PCA_large <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#4292c6"</span>, <span class="st">"#41ab5d"</span>, <span class="st">"#9e9ac8"</span>,  <span class="st">"#fd8d3c"</span>, <span class="st">"#df65b0"</span>,<span class="st">"#9ecae1"</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>number_of_genes.color <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#4292c6"</span>, <span class="st">"#41ab5d"</span>, <span class="st">"#9e9ac8"</span>,  <span class="st">"#fd8d3c"</span>, <span class="st">"#df65b0"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>theme_GE <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">base_size =</span> <span class="dv">15</span>, <span class="at">base_family =</span> <span class="st">""</span>) {</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> base_size, <span class="at">base_family =</span> base_family) <span class="sc">+</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Set the overall text size and font</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size, <span class="at">family =</span> base_family),</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Modify title text</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Ensure axis titles are shown</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size),</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size),</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Ensure axis ticks and texts are visible</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size),</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>), <span class="co"># Make ticks black</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Keep the background of the plot area blank</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Keep panel background blank</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Modify legend position</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">"right"</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Modify plot margin</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>      <span class="co">#plot.margin = margin(t = 5, r = 5, b = 5, l = 5, unit = "pt")</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>theme_gut <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">base_size =</span> <span class="dv">9</span>, <span class="at">base_family =</span> <span class="st">"arial"</span>) {</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> base_size, <span class="at">base_family =</span> base_family) <span class="sc">+</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">colour =</span> <span class="cn">NA</span>),</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">0.3</span>),</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">0.3</span>),</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size <span class="sc">+</span> <span class="dv">1</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size),</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">1</span>),</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">1</span>),</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size),</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size, <span class="at">face =</span> <span class="st">"bold"</span>)</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>capitalize_first <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(x, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(x, <span class="dv">2</span>))</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the correct tissue levels (based on your list)</span></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>tissue_levels_all <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Duodenum"</span>,</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Treitz"</span>,</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 3"</span>,</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 4"</span>,</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 5"</span>,</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 6"</span>,</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 7"</span>,</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 8"</span>,</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 9"</span>,</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Ileocecal"</span>,</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Cecum"</span>,</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Ascending colon"</span>,</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Transverse colon"</span>,</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Descending colon"</span>,</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Sigmoideum"</span>,</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Rectum"</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-raw-datafiles-process-df-and-design" class="level2">
<h2 class="anchored" data-anchor-id="load-raw-datafiles-process-df-and-design">Load raw datafiles &amp; Process df and design</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Load raw datafiles</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"/Users/hannahgilliam-vigh/Documents/Publicationer/Dataset/Data_NotNorm/20220502_Counts.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">dec =</span> <span class="st">"."</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>design_df <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"//Users/hannahgilliam-vigh/Documents/Publicationer/Dataset/Data_NotNorm/20220502_Design.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">dec =</span> <span class="st">"."</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>ProtCod <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"/Users/hannahgilliam-vigh/Documents/Publicationer/Dataset/Data_NotNorm/20220502_PC.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">dec =</span> <span class="st">"."</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>RPKM_Reads <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"/Users/hannahgilliam-vigh/Documents/Publicationer/Dataset/GUS2018-544-GH RNAseq gut RNAseq tables.xlsx"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="do">##Process df and design</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Initial dimensions of df and design_df</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Initial dimensions of df:"</span>, <span class="fu">dim</span>(df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Initial dimensions of df: 57722 357 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Initial dimensions of design_df:"</span>, <span class="fu">dim</span>(design_df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Initial dimensions of design_df: 357 31 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust column names in 'df' by removing the first character and keeping positions 2 to 5</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">colnames</span>(df), <span class="dv">2</span>, <span class="dv">5</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the first five column names as a check</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"First 5 column names in df after renaming:"</span>, <span class="fu">colnames</span>(df)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>First 5 column names in df after renaming: 6204 6205 6206 6207 6208 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the first five row names of 'design_df' to verify alignment</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"First 5 row names in design_df:"</span>, <span class="fu">rownames</span>(design_df)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>First 5 row names in design_df: 6204 6205 6206 6207 6208 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if row names of 'design_df' match column names of 'df'</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>alignment_check <span class="ot">&lt;-</span> <span class="fu">all</span>(<span class="fu">rownames</span>(design_df) <span class="sc">==</span> <span class="fu">colnames</span>(df))</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Are row names of design_df and col names of df identical?"</span>, alignment_check, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Are row names of design_df and col names of df identical? TRUE </code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dimensions before filtering</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Dimensions before filtering:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Dimensions before filtering:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"df:"</span>, <span class="fu">dim</span>(df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>df: 57722 357 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"design_df:"</span>, <span class="fu">dim</span>(design_df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>design_df: 357 31 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define known outlier samples to exclude</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"6503"</span>, <span class="st">"6462"</span>, <span class="st">"6464"</span>, <span class="st">"6463"</span>, <span class="st">"6299"</span>, <span class="st">"6313"</span>, <span class="st">"6351"</span>, <span class="st">"6374"</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>outlier_info <span class="ot">&lt;-</span> design_df <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">rownames</span>(design_df) <span class="sc">%in%</span> outliers)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove specific samples from 'design_df' that are known outliers or should be excluded</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>design_df <span class="ot">&lt;-</span> design_df[<span class="sc">!</span><span class="fu">rownames</span>(design_df) <span class="sc">%in%</span> outliers,]</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Dimensions after filtering</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Dimensions after filtering design_df:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Dimensions after filtering design_df:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"design_df:"</span>, <span class="fu">dim</span>(design_df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>design_df: 349 31 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset 'df' to retain only the columns that correspond to the updated 'design_df'</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df[, <span class="fu">rownames</span>(design_df) ]</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Final dimensions after subsetting df</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Final dimensions of df:"</span>, <span class="fu">dim</span>(df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Final dimensions of df: 57722 349 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">## New columns - New categorization based on heatmap </span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a unique ID for each sample by combining Tissue and Treatment</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>design_df<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">paste0</span>(design_df<span class="sc">$</span>Tissue, <span class="st">"_"</span>, design_df<span class="sc">$</span>Treatment))</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Categorize samples into broader anatomical locations</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>design_df <span class="ot">&lt;-</span> design_df <span class="sc">%&gt;%</span> </span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Location =</span> <span class="fu">case_when</span>(</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Duodenum"</span>, <span class="st">"Treitz"</span>, <span class="st">"small intestine 3"</span>) <span class="sc">~</span> <span class="st">"ProxSmall"</span>,</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"small intestine 4"</span>, <span class="st">"small intestine 5"</span>, <span class="st">"small intestine 6"</span>) <span class="sc">~</span> <span class="st">"MidSmall"</span>,</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"small intestine 7"</span>, <span class="st">"small intestine 8"</span>, <span class="st">"small intestine 9"</span>, <span class="st">"ileocecal"</span>) <span class="sc">~</span> <span class="st">"DistSmall"</span>,</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"cecum"</span>, <span class="st">"ascending colon"</span>, <span class="st">"transverse colon"</span>) <span class="sc">~</span> <span class="st">"ProxLarge"</span>,</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"descending colon"</span>, <span class="st">"sigmoideum"</span>, <span class="st">"rectum"</span>) <span class="sc">~</span> <span class="st">"DistLarge"</span>,</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"other"</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign numerical labels to tissue locations for ordered analysis</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>design_df <span class="ot">&lt;-</span> design_df <span class="sc">%&gt;%</span> </span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Location.num =</span> <span class="fu">case_when</span>(</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"Duodenum"</span> <span class="sc">~</span> <span class="st">"1"</span>,</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"Treitz"</span> <span class="sc">~</span> <span class="st">"2"</span>,</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"small intestine 3"</span> <span class="sc">~</span> <span class="st">"3"</span>,</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"small intestine 4"</span> <span class="sc">~</span> <span class="st">"4"</span>,</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"small intestine 5"</span> <span class="sc">~</span> <span class="st">"5"</span>,</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"small intestine 6"</span> <span class="sc">~</span> <span class="st">"6"</span>,</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"small intestine 7"</span> <span class="sc">~</span> <span class="st">"7"</span>,</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"small intestine 8"</span> <span class="sc">~</span> <span class="st">"8"</span>,</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"small intestine 9"</span> <span class="sc">~</span> <span class="st">"9"</span>,</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"ileocecal"</span> <span class="sc">~</span> <span class="st">"10"</span>,</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"cecum"</span> <span class="sc">~</span> <span class="st">"11"</span>,</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"ascending colon"</span> <span class="sc">~</span> <span class="st">"12"</span>,</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"transverse colon"</span> <span class="sc">~</span> <span class="st">"13"</span>,</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"descending colon"</span> <span class="sc">~</span> <span class="st">"14"</span>,</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"sigmoideum"</span> <span class="sc">~</span> <span class="st">"15"</span>,</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>    Tissue <span class="sc">==</span> <span class="st">"rectum"</span> <span class="sc">~</span> <span class="st">"16"</span>,</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"other"</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure only the first letter of the string is capitalized</span></span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>design_df <span class="ot">&lt;-</span> design_df <span class="sc">%&gt;%</span></span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Tissue =</span> <span class="fu">str_to_sentence</span>(Tissue),  <span class="co"># Converts "small intestine 3" -&gt; "Small intestine 3"</span></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">Subject =</span> <span class="fu">case_when</span>(</span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>      Treatment <span class="sc">==</span> <span class="st">"T2DM"</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"D"</span>, Subject),</span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>      Treatment <span class="sc">==</span> <span class="st">"Kontrol"</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"K"</span>, Subject),</span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">as.character</span>(Subject)  <span class="co"># fallback, just in case</span></span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">Tissue =</span> <span class="fu">factor</span>(Tissue, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Duodenum"</span>,<span class="st">"Treitz"</span>,<span class="st">"Small intestine 3"</span>, <span class="st">"Small intestine 4"</span>, <span class="st">"Small intestine 5"</span>, <span class="st">"Small intestine 6"</span>, <span class="st">"Small intestine 7"</span>,<span class="st">"Small intestine 8"</span>, <span class="st">"Small intestine 9"</span>,<span class="st">"Ileocecal"</span>, <span class="st">"Cecum"</span>, <span class="st">"Ascending colon"</span>, <span class="st">"Transverse colon"</span>, </span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Descending colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>))</span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Check updated dimensions after adding new columns</span></span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Final dimensions of design_df after adding new columns:"</span>, <span class="fu">dim</span>(design_df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Final dimensions of design_df after adding new columns: 349 34 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter df based on protein-coding genes</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Check current dimensions of df and ProtCod before filtering</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Dimensions of df before filtering for protein-coding genes:"</span>, <span class="fu">dim</span>(df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Dimensions of df before filtering for protein-coding genes: 57722 349 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Dimensions of ProtCod:"</span>, <span class="fu">dim</span>(ProtCod), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Dimensions of ProtCod: 19786 1 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only rows in df that correspond to protein-coding genes in ProtCod</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df[<span class="fu">rownames</span>(df) <span class="sc">%in%</span> ProtCod<span class="sc">$</span>ensembl_gene_id, ]</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check dimensions after filtering</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Dimensions of df after filtering for protein-coding genes:"</span>, <span class="fu">dim</span>(df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Dimensions of df after filtering for protein-coding genes: 19739 349 </code></pre>
</div>
</div>
<section id="rpkm-and-vst-data-annotation" class="level3">
<h3 class="anchored" data-anchor-id="rpkm-and-vst-data-annotation">RPKM and VST data annotation</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Define location</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>location_mapping <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Duodenum"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"Treitz"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"small intestine 3"</span> <span class="ot">=</span> <span class="dv">3</span>,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"small intestine 4"</span> <span class="ot">=</span> <span class="dv">4</span>, <span class="st">"small intestine 5"</span> <span class="ot">=</span> <span class="dv">5</span>,</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"small intestine 6"</span> <span class="ot">=</span> <span class="dv">6</span>, <span class="st">"small intestine 7"</span> <span class="ot">=</span> <span class="dv">7</span>,</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"small intestine 8"</span> <span class="ot">=</span> <span class="dv">8</span>, <span class="st">"small intestine 9"</span> <span class="ot">=</span> <span class="dv">9</span>, </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ileocecal"</span> <span class="ot">=</span> <span class="dv">10</span>, <span class="st">"cecum"</span> <span class="ot">=</span> <span class="dv">11</span>, <span class="st">"ascending colon"</span><span class="ot">=</span> <span class="dv">12</span>, </span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"transverse colon"</span> <span class="ot">=</span> <span class="dv">13</span>, <span class="st">"descending colon"</span> <span class="ot">=</span> <span class="dv">14</span>, </span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"sigmoideum"</span> <span class="ot">=</span> <span class="dv">15</span>, <span class="st">"rectum"</span> <span class="ot">=</span> <span class="dv">16</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Gather long format</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>rpkm_long <span class="ot">&lt;-</span> RPKM_Reads <span class="sc">%&gt;%</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">Ensembl acc key</span><span class="st">`</span>, <span class="st">`</span><span class="at">Gene name</span><span class="st">`</span>, Description),</span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"Expression"</span></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: Add metadata</span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>rpkm_long <span class="ot">&lt;-</span> rpkm_long <span class="sc">%&gt;%</span></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">Type =</span> <span class="fu">ifelse</span>(<span class="fu">str_detect</span>(Sample, <span class="st">"Kontrol"</span>), <span class="st">"Kontrol"</span>, <span class="st">"T2DM"</span>),</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">Subject =</span> <span class="fu">str_extract</span>(Sample, <span class="st">"animal </span><span class="sc">\\</span><span class="st">d+"</span>) <span class="sc">%&gt;%</span> <span class="fu">str_replace</span>(<span class="st">"animal "</span>, <span class="st">""</span>),</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">Subject =</span> <span class="fu">ifelse</span>(Type <span class="sc">==</span> <span class="st">"Kontrol"</span>, <span class="fu">paste0</span>(<span class="st">"K"</span>, Subject), <span class="fu">paste0</span>(<span class="st">"D"</span>, Subject)),</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">Tissue_raw =</span> <span class="fu">str_extract</span>(Sample, <span class="st">"Duodenum|Treitz|small intestine 3|small intestine 4|small intestine 5|small intestine 6|small intestine 7|small intestine 8|small intestine 9|ileocecal|cecum|ascending.colon|transverse.colon|descending.colon|sigmoideum|rectum"</span>),</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">Location.num =</span> location_mapping[Tissue_raw]</span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 3: Add factors and clean column names</span></span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>rpkm_long <span class="ot">&lt;-</span> rpkm_long <span class="sc">%&gt;%</span></span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Location.num)) <span class="sc">%&gt;%</span></span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">Location =</span> <span class="fu">factor</span>(Location.num, <span class="at">levels =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">16</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'Duodenum'</span>, <span class="st">'Treitz'</span>,<span class="st">'3'</span>,<span class="st">'4'</span>,<span class="st">'5'</span>,<span class="st">'6'</span>,<span class="st">'7'</span>,<span class="st">'8'</span>,<span class="st">'9'</span>,<span class="st">'Ileocecal'</span>,<span class="st">"Cecum"</span>, <span class="st">"Asc. colon"</span>, <span class="st">"Trans. colon"</span>, <span class="st">"Desc. colon"</span>, <span class="st">"Sigmoid colon"</span>, <span class="st">"Rectum"</span>)),</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">Type =</span> <span class="fu">factor</span>(Type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Kontrol"</span>, <span class="st">"T2DM"</span>))</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(</span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">ENSEMBL =</span> <span class="st">`</span><span class="at">Ensembl acc key</span><span class="st">`</span>,</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">SYMBOL =</span> <span class="st">`</span><span class="at">Gene name</span><span class="st">`</span>,</span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">GENENAME =</span> Description</span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(ENSEMBL, SYMBOL, GENENAME, Expression, Type, Location, Subject)</span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 4: Annotate with ENTREZID</span></span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a>gene_map <span class="ot">&lt;-</span> <span class="fu">bitr</span>(</span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>(rpkm_long<span class="sc">$</span>ENSEMBL),</span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a>  <span class="at">toType   =</span> <span class="fu">c</span>(<span class="st">"ENTREZID"</span>),</span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb    =</span> org.Hs.eg.db</span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bitr(unique(rpkm_long$ENSEMBL), fromType = "ENSEMBL", toType =
c("ENTREZID"), : 2.59% of input gene IDs are fail to map...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>rpkm_annotated <span class="ot">&lt;-</span> rpkm_long <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gene_map, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">entrezgene_id =</span> ENTREZID) <span class="sc">%&gt;%</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(entrezgene_id))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., gene_map, by = "ENSEMBL"): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 82477 of `x` matches multiple rows in `y`.
ℹ Row 1 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
</div>
</section>
</section>
<section id="pca-of-all-intestinal-segments" class="level2">
<h2 class="anchored" data-anchor-id="pca-of-all-intestinal-segments">PCA of All Intestinal Segments</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Prepare data</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - 'design.' is the sample metadata (experimental design)</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - 'count.' is the gene expression count matrix</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> design_df</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>count. <span class="ot">&lt;-</span> df</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: Create a DESeq2 dataset</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co"># - This normalizes counts while accounting for differences in sequencing depth</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>dds. <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(count., design., <span class="at">design =</span> <span class="sc">~</span>ID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 3: Variance Stabilizing Transformation (VST)</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - VST is applied to normalize data and reduce heteroskedasticity</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - 'vst_values.' contains the transformed expression values</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>vst_values. <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds.) <span class="sc">%&gt;%</span> <span class="fu">assay</span>()</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Selecting the Most Variable Genes for PCA</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 4: Calculate gene variability</span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="co"># - Compute row-wise variance for each gene to identify the most variable genes</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>tmp. <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">ensembl_gene_id =</span> <span class="fu">rownames</span>(vst_values.),</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">rowVar =</span> <span class="fu">rowVars</span>(vst_values.)</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 5: Rank genes by variance (descending order)</span></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a><span class="co"># - Higher variance indicates greater variation across samples</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>tmp. <span class="ot">&lt;-</span> tmp.[<span class="fu">order</span>(tmp.<span class="sc">$</span>rowVar, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),]</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 6: Select the top 500 most variable genes for PCA</span></span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>top_x <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>use.these.<span class="fl">500.</span> <span class="ot">&lt;-</span> tmp.<span class="sc">$</span>ensembl_gene_id[<span class="dv">1</span><span class="sc">:</span>top_x]</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 7: Subset vst_values to include only the top 500 most variable genes</span></span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>my.pca. <span class="ot">&lt;-</span> vst_values.[use.these.<span class="fl">500.</span>, <span class="fu">rownames</span>(design.)]</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a><span class="do">## Performing PCA</span></span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 8: Perform Principal Component Analysis (PCA)</span></span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a><span class="co"># - Transpose data since PCA expects samples in rows and genes in columns</span></span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>my.pca. <span class="ot">&lt;-</span> FactoMineR<span class="sc">::</span><span class="fu">PCA</span>(<span class="fu">t</span>(my.pca.))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Preparing Data for Visualization</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 10: Extract PCA coordinates</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co"># - This dataframe contains the position of each sample in the PCA space</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>plot.this. <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(my.pca.<span class="sc">$</span>ind<span class="sc">$</span>coord)</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 11: Merge PCA coordinates with metadata</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="co"># - Adds tissue type, treatment, and sample information</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>plot.this. <span class="ot">&lt;-</span> <span class="fu">merge</span>(plot.this., design.[,<span class="fu">c</span>(<span class="st">"Tissue"</span>, <span class="st">"Treatment"</span>, <span class="st">"Sample_run"</span>, <span class="st">"Subject"</span>)], <span class="at">by=</span><span class="st">"row.names"</span>)</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 12: Compute mean PCA coordinates per tissue type</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>plot.this.sum. <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">group_by</span>(plot.this., Tissue)</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>plot.this.sum. <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(plot.this.sum., <span class="fu">mean</span>(Dim<span class="fl">.1</span>), <span class="fu">mean</span>(Dim<span class="fl">.2</span>), <span class="fu">mean</span>(Dim<span class="fl">.3</span>)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 13: Rename columns for clarity</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(plot.this.sum.) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Tissue"</span>, <span class="st">"Dim.1"</span>, <span class="st">"Dim.2"</span>, <span class="st">"Dim.3"</span>)</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 14: Select data for T2DM treatment</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>plot.this.t2dm <span class="ot">&lt;-</span> plot.this.[plot.this.<span class="sc">$</span>Treatment <span class="sc">==</span> <span class="st">"T2DM"</span>,]</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Creating the PCA Plot</span></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 15: Set ggplot options</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">ggrepel.max.overlaps =</span> <span class="cn">Inf</span>)  <span class="co"># Prevent excessive label overlaps</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 16: Generate PCA plot using ggplot2</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="cn">NULL</span>, <span class="fu">aes</span>((Dim<span class="fl">.1</span>) <span class="sc">*</span> <span class="sc">-</span><span class="dv">1</span>, (Dim<span class="fl">.2</span>) <span class="sc">*</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">bg =</span> Tissue)) <span class="sc">+</span></span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add labels for tissue clusters (mean PCA coordinates)</span></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="at">data =</span> plot.this.sum., </span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">aes</span>(<span class="at">point.size =</span> <span class="dv">4</span>, <span class="at">x =</span> (Dim<span class="fl">.1</span>) <span class="sc">*</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">y =</span> (Dim<span class="fl">.2</span>) <span class="sc">*</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">label =</span> Tissue), </span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">box.padding =</span> <span class="dv">1</span>, <span class="at">label.padding =</span> <span class="fl">0.2</span>, <span class="at">point.padding =</span> <span class="dv">45</span>, </span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">max.overlaps =</span> <span class="cn">Inf</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add points for T2DM samples (black dots)</span></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> plot.this.t2dm, <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add points for all samples</span></span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> plot.this., <span class="at">size =</span> <span class="dv">4</span>, <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply theme</span></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_tufte</span>(<span class="at">base_family =</span> <span class="st">"Arial"</span>, <span class="at">base_size =</span> <span class="dv">11</span>) <span class="sc">+</span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.4</span>)) <span class="sc">+</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Dynamically generate axis labels using PCA variance explained</span></span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC 1 ("</span>, <span class="fu">round</span>(my.pca.<span class="sc">$</span>eig[<span class="dv">1</span>, <span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"%)"</span>)) <span class="sc">+</span></span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC 2 ("</span>, <span class="fu">round</span>(my.pca.<span class="sc">$</span>eig[<span class="dv">2</span>, <span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"%)"</span>)) <span class="sc">+</span></span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply tissue color scheme</span></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> PCA_all)</span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 17: Display the PCA plot</span></span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = p, "PCA_plot_all", width = 20.37/2.5, height = 11.99/2.5)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pca-of-small-intestine-samples" class="level2">
<h2 class="anchored" data-anchor-id="pca-of-small-intestine-samples">PCA of Small Intestine Samples</h2>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/small%20PCA-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/small%20PCA-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/small%20PCA-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-of-large-intestine-samples" class="level2">
<h2 class="anchored" data-anchor-id="pca-of-large-intestine-samples">PCA of Large Intestine Samples</h2>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/large%20PCA-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/large%20PCA-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/large%20PCA-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="spatial-differences-compared-to-duodenum-or-rectum-overview" class="level1">
<h1>Spatial differences compared to Duodenum or Rectum Overview</h1>
<p>This script analyzes spatial differences in gene expression along the small and large intestine by comparing each intestinal segment to a reference tissue (Duodenum for the small intestine, Rectum for the large intestine).</p>
<p>The script:</p>
<ul>
<li>Filters control samples (Kontrol) for spatial comparison.</li>
<li>Runs DESeq2 to identify differentially expressed genes (DEGs).</li>
<li>Computes variance-stabilized transformed (VST) values for visualization.</li>
<li>Extracts significantly regulated genes (padj &lt; 0.05) for each segment.</li>
<li>Computes mean expression values for each tissue.</li>
</ul>
<p>This script takes a long time to run, and results have already been saved.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 1: Filter Small Intestine Samples</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - Select only samples from the small intestine (Location.num &lt; 11).</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - This ensures we are analyzing only the small intestine while excluding large intestine segments.</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> design_df </span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>design.<span class="sc">$</span>Tissue <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(design.<span class="sc">$</span>Tissue) <span class="sc">%&gt;%</span> <span class="fu">relevel</span>(design.<span class="sc">$</span>Tissue, <span class="at">ref =</span> <span class="st">"Duodenum"</span>)</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> design.[design.<span class="sc">$</span>Treatment <span class="sc">==</span> <span class="st">"Kontrol"</span>,]</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> design.[design.<span class="sc">$</span>Treatment <span class="sc">==</span> <span class="st">"T2DM"</span>,]</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> <span class="fu">subset</span>(design_df, <span class="fu">as.numeric</span>(Location.num) <span class="sc">&lt;</span> <span class="dv">11</span>) </span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>design.<span class="sc">$</span>Tissue <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(design.<span class="sc">$</span>Tissue) <span class="sc">%&gt;%</span> <span class="fu">relevel</span>(design.<span class="sc">$</span>Tissue, <span class="at">ref =</span> <span class="st">"Duodenum"</span>)</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> <span class="fu">subset</span>(design_df, <span class="fu">as.numeric</span>(Location.num) <span class="sc">&gt;</span> <span class="dv">10</span>) </span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>design.<span class="sc">$</span>Tissue <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(design.<span class="sc">$</span>Tissue) <span class="sc">%&gt;%</span> <span class="fu">relevel</span>(design.<span class="sc">$</span>Tissue, <span class="at">ref =</span> <span class="st">"Rectum"</span>)</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>design.<span class="sc">$</span>Treatment <span class="ot">&lt;-</span> <span class="fu">factor</span>(design.<span class="sc">$</span>Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Kontrol"</span>,<span class="st">"T2DM"</span>))</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the corresponding count matrix (gene expression data) for selected samples</span></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>count. <span class="ot">&lt;-</span> df[, <span class="fu">rownames</span>(design.)]</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 2: Filter for Control (Kontrol) Samples</span></span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a><span class="co"># - Only use "Kontrol" samples to assess spatial variation.</span></span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a><span class="co"># - This avoids confounding treatment effects from T2DM.</span></span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>design.control <span class="ot">&lt;-</span> design.[design.<span class="sc">$</span>Treatment <span class="sc">==</span> <span class="st">"Kontrol"</span>,]</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>counts.control <span class="ot">&lt;-</span> df[, <span class="fu">rownames</span>(design.control)]</span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 3: Prepare DESeq2 Dataset &amp; VST Transformation</span></span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a><span class="co"># - Normalize data using DESeq2's size factor estimation.</span></span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a><span class="co"># - Apply Variance Stabilizing Transformation (VST) for downstream analysis.</span></span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a><span class="co">#dds. &lt;- DESeqDataSetFromMatrix(counts.control, design.control, design = ~Tissue)</span></span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a>dds. <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(count., design., <span class="at">design =</span> <span class="sc">~</span>Tissue)</span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a>dds. <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds.)</span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a>vst_values. <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds.) <span class="sc">%&gt;%</span> <span class="fu">assay</span>()</span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a>dds. <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds.)  <span class="co"># Run DESeq2 model</span></span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(dds.)   <span class="co"># Check available contrast names</span></span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 4: Define Small Intestine Segments (Excluding Duodenum)</span></span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a><span class="co">#All Segments </span></span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>segments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Treitz"</span>, <span class="st">"Small.intestine.3"</span>, <span class="st">"Small.intestine.4"</span>, <span class="st">"Small.intestine.5"</span>, <span class="st">"Small.intestine.6"</span>, <span class="st">"Small.intestine.7"</span>, </span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small.intestine.8"</span>, <span class="st">"Small.intestine.9"</span>, <span class="st">"Ileocecal"</span>,<span class="st">"Cecum"</span>, <span class="st">"Ascending.colon"</span>, <span class="st">"Transverse.colon"</span>,<span class="st">"Descending.colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>)</span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a><span class="co">#Small Intestines</span></span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a>segments <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Treitz"</span>, <span class="st">"Small.intestine.3"</span>, <span class="st">"Small.intestine.4"</span>, </span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small.intestine.5"</span>, <span class="st">"Small.intestine.6"</span>, <span class="st">"Small.intestine.7"</span>, </span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small.intestine.8"</span>, <span class="st">"Small.intestine.9"</span>, <span class="st">"Ileocecal"</span></span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a><span class="co">#Large Intestines</span></span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a>segments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Cecum"</span>, <span class="st">"Ascending.colon"</span>, <span class="st">"Transverse.colon"</span>,<span class="st">"Descending.colon"</span>, <span class="st">"Sigmoideum"</span>)</span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-55"><a href="#cb46-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 5: Compute Differential Expression Relative to Reference segment</span></span>
<span id="cb46-56"><a href="#cb46-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Example - Extracts genes differentially expressed in each small intestine part vs. Duodenum.</span></span>
<span id="cb46-57"><a href="#cb46-57" aria-hidden="true" tabindex="-1"></a>results.list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(segments, <span class="cf">function</span>(x){</span>
<span id="cb46-58"><a href="#cb46-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-59"><a href="#cb46-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct contrast name based on DESeq2 resultsNames(dds.)</span></span>
<span id="cb46-60"><a href="#cb46-60" aria-hidden="true" tabindex="-1"></a>  contrast_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Tissue_"</span>, x, <span class="st">"_vs_Duodenum"</span>)  </span>
<span id="cb46-61"><a href="#cb46-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-62"><a href="#cb46-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if contrast exists before running results() to prevent errors</span></span>
<span id="cb46-63"><a href="#cb46-63" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (contrast_name <span class="sc">%in%</span> <span class="fu">resultsNames</span>(dds.)) {</span>
<span id="cb46-64"><a href="#cb46-64" aria-hidden="true" tabindex="-1"></a>    tmp.res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds., <span class="at">name =</span> contrast_name)</span>
<span id="cb46-65"><a href="#cb46-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tmp.res)</span>
<span id="cb46-66"><a href="#cb46-66" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb46-67"><a href="#cb46-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Contrast not found for: "</span>, x)  <span class="co"># Debugging message</span></span>
<span id="cb46-68"><a href="#cb46-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="cn">NULL</span>)  <span class="co"># Avoids breaking the loop if contrast is missing</span></span>
<span id="cb46-69"><a href="#cb46-69" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb46-70"><a href="#cb46-70" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb46-71"><a href="#cb46-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-72"><a href="#cb46-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign the tissue names to the results list</span></span>
<span id="cb46-73"><a href="#cb46-73" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(results.list) <span class="ot">&lt;-</span> segments</span>
<span id="cb46-74"><a href="#cb46-74" aria-hidden="true" tabindex="-1"></a><span class="co"># View summary statistics for one of the results (e.g., Treitz)</span></span>
<span id="cb46-75"><a href="#cb46-75" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(results.list)</span>
<span id="cb46-76"><a href="#cb46-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-77"><a href="#cb46-77" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 6: Extract Significantly Regulated Genes (`padj &lt; 0.05`)</span></span>
<span id="cb46-78"><a href="#cb46-78" aria-hidden="true" tabindex="-1"></a><span class="co"># - Retain only differentially expressed genes with an adjusted p-value below 0.05.</span></span>
<span id="cb46-79"><a href="#cb46-79" aria-hidden="true" tabindex="-1"></a>regulated.genes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(results.list, <span class="cf">function</span>(x){</span>
<span id="cb46-80"><a href="#cb46-80" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(x)  <span class="co"># Remove NAs</span></span>
<span id="cb46-81"><a href="#cb46-81" aria-hidden="true" tabindex="-1"></a>  tmp_padj_genes <span class="ot">&lt;-</span> x[x<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>,]  <span class="co"># Select significantly differentially expressed genes</span></span>
<span id="cb46-82"><a href="#cb46-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">as.data.frame</span>(tmp_padj_genes))  <span class="co"># Convert to a data frame</span></span>
<span id="cb46-83"><a href="#cb46-83" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb46-84"><a href="#cb46-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-85"><a href="#cb46-85" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign tissue names to the list of regulated genes</span></span>
<span id="cb46-86"><a href="#cb46-86" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(regulated.genes) <span class="ot">&lt;-</span> segments</span>
<span id="cb46-87"><a href="#cb46-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-88"><a href="#cb46-88" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 7: Extract Unique Regulated Genes Across All Intestinal Segments</span></span>
<span id="cb46-89"><a href="#cb46-89" aria-hidden="true" tabindex="-1"></a><span class="co"># - Unlists all regulated genes, removes duplicates, and stores them in a vector.</span></span>
<span id="cb46-90"><a href="#cb46-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-91"><a href="#cb46-91" aria-hidden="true" tabindex="-1"></a>regulated.genes. <span class="ot">&lt;-</span> <span class="fu">lapply</span>(regulated.genes, <span class="cf">function</span>(x){</span>
<span id="cb46-92"><a href="#cb46-92" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">row.names</span>(x)  <span class="co"># Extract gene names</span></span>
<span id="cb46-93"><a href="#cb46-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb46-94"><a href="#cb46-94" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> </span>
<span id="cb46-95"><a href="#cb46-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>(., <span class="at">use.names =</span> F) <span class="sc">%&gt;%</span>  <span class="co"># Flatten list into a single vector</span></span>
<span id="cb46-96"><a href="#cb46-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()  <span class="co"># Remove duplicates</span></span>
<span id="cb46-97"><a href="#cb46-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-98"><a href="#cb46-98" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 8: Define Ordered Tissue Labels for Visualization</span></span>
<span id="cb46-99"><a href="#cb46-99" aria-hidden="true" tabindex="-1"></a><span class="co"># - Used to organize tissues in spatial order for downstream analysis.</span></span>
<span id="cb46-100"><a href="#cb46-100" aria-hidden="true" tabindex="-1"></a>segments <span class="ot">&lt;-</span> <span class="fu">c</span>(  <span class="st">"Duodenum"</span>, <span class="st">"Treitz"</span>, <span class="st">"Small intestine 3"</span>, <span class="st">"Small intestine 4"</span>, <span class="st">"Small intestine 5"</span>, <span class="st">"Small intestine 6"</span>, <span class="st">"Small intestine 7"</span>,</span>
<span id="cb46-101"><a href="#cb46-101" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 8"</span>, <span class="st">"Small intestine 9"</span>, <span class="st">"Ileocecal"</span>, <span class="st">"Cecum"</span>, <span class="st">"Ascending colon"</span>, <span class="st">"Transverse colon"</span>,<span class="st">"Descending colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>)</span>
<span id="cb46-102"><a href="#cb46-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-103"><a href="#cb46-103" aria-hidden="true" tabindex="-1"></a>segments <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="st">"Duodenum"</span>, <span class="st">"Treitz"</span>, <span class="st">"Small intestine 3"</span>, <span class="st">"Small intestine 4"</span>, <span class="st">"Small intestine 5"</span>, <span class="st">"Small intestine 6"</span>, <span class="st">"Small intestine 7"</span>,</span>
<span id="cb46-104"><a href="#cb46-104" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Small intestine 8"</span>, <span class="st">"Small intestine 9"</span>, <span class="st">"Ileocecal"</span>)</span>
<span id="cb46-105"><a href="#cb46-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-106"><a href="#cb46-106" aria-hidden="true" tabindex="-1"></a>segments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Cecum"</span>, <span class="st">"Ascending colon"</span>, <span class="st">"Transverse colon"</span>,<span class="st">"Descending colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>)</span>
<span id="cb46-107"><a href="#cb46-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-108"><a href="#cb46-108" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 9: Group Samples by Tissue Type</span></span>
<span id="cb46-109"><a href="#cb46-109" aria-hidden="true" tabindex="-1"></a><span class="co"># - Creates a mapping of sample IDs to their corresponding tissue.</span></span>
<span id="cb46-110"><a href="#cb46-110" aria-hidden="true" tabindex="-1"></a><span class="co"># - Useful for plotting and averaging expression levels.</span></span>
<span id="cb46-111"><a href="#cb46-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-112"><a href="#cb46-112" aria-hidden="true" tabindex="-1"></a>together.by.tissue. <span class="ot">&lt;-</span> <span class="fu">lapply</span>(segments, <span class="cf">function</span>(x){ </span>
<span id="cb46-113"><a href="#cb46-113" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> design. <span class="sc">%&gt;%</span> </span>
<span id="cb46-114"><a href="#cb46-114" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Tissue <span class="sc">==</span> x) <span class="sc">%&gt;%</span></span>
<span id="cb46-115"><a href="#cb46-115" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Sample_run)  <span class="co"># Extracts sample IDs for each tissue</span></span>
<span id="cb46-116"><a href="#cb46-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb46-117"><a href="#cb46-117" aria-hidden="true" tabindex="-1"></a>}) </span>
<span id="cb46-118"><a href="#cb46-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-119"><a href="#cb46-119" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(together.by.tissue.) <span class="ot">&lt;-</span> segments</span>
<span id="cb46-120"><a href="#cb46-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-121"><a href="#cb46-121" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 10: Compute Mean Expression per Tissue</span></span>
<span id="cb46-122"><a href="#cb46-122" aria-hidden="true" tabindex="-1"></a><span class="co"># - Uses variance-stabilized (VST) values to compute average expression per tissue.</span></span>
<span id="cb46-123"><a href="#cb46-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-124"><a href="#cb46-124" aria-hidden="true" tabindex="-1"></a>mean.by.tissue. <span class="ot">&lt;-</span> <span class="fu">lapply</span>(together.by.tissue., <span class="cf">function</span>(x){</span>
<span id="cb46-125"><a href="#cb46-125" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> vst_values.[, <span class="fu">colnames</span>(vst_values.) <span class="sc">%in%</span> x<span class="sc">$</span>Sample_run]  <span class="co"># Select matching samples</span></span>
<span id="cb46-126"><a href="#cb46-126" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(tmp)</span>
<span id="cb46-127"><a href="#cb46-127" aria-hidden="true" tabindex="-1"></a>  tmp<span class="sc">$</span>mean <span class="ot">&lt;-</span> <span class="fu">apply</span>(tmp, <span class="dv">1</span>, mean)  <span class="co"># Compute row-wise mean expression</span></span>
<span id="cb46-128"><a href="#cb46-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">as.data.frame</span>(tmp<span class="sc">$</span>mean))  <span class="co"># Convert to dataframe</span></span>
<span id="cb46-129"><a href="#cb46-129" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb46-130"><a href="#cb46-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-131"><a href="#cb46-131" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 11: Combine Results into a Single Matrix</span></span>
<span id="cb46-132"><a href="#cb46-132" aria-hidden="true" tabindex="-1"></a><span class="co"># - Converts the list of means into a dataframe with one column per tissue.</span></span>
<span id="cb46-133"><a href="#cb46-133" aria-hidden="true" tabindex="-1"></a>mean.by.tissue. <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, mean.by.tissue.)</span>
<span id="cb46-134"><a href="#cb46-134" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mean.by.tissue.) <span class="ot">&lt;-</span> <span class="fu">names</span>(together.by.tissue.)</span>
<span id="cb46-135"><a href="#cb46-135" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mean.by.tissue.) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(vst_values.)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="heatmap-small" class="level2">
<h2 class="anchored" data-anchor-id="heatmap-small">Heatmap Small</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 1: Load Previously Saved Data</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - These files contain the results from the previous analysis</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - 'regulated.genes_small_control.Rdata' holds the list of significantly regulated genes (padj &lt; 0.05)</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co"># - 'mean.by.tissue_small_control.Rdata' contains mean gene expression values per tissue (VST transformed)</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"regulated.genes_small_control.Rdata"</span>)   <span class="co"># Load significantly regulated genes</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"mean.by.tissue_small_control.Rdata"</span>)    <span class="co"># Load mean expression data for each tissue</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 2: Define Number of Clusters for Heatmap</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a><span class="co"># - This determines how the hierarchical clustering will be cut into distinct clusters</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>number_of_clusters <span class="ot">&lt;-</span> <span class="dv">8</span>  <span class="co"># Adjust as needed</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 3: Filter Expression Data for Significantly Regulated Genes</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a><span class="co"># - Extract only the genes from 'mean.by.tissue.' that are in the 'regulated_genes.' list</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>heat.cluster. <span class="ot">&lt;-</span> mean.by.tissue.[<span class="fu">rownames</span>(mean.by.tissue.) <span class="sc">%in%</span> regulated_genes., ]</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 4: Generate Heatmap to Visualise Spatial Expression Patterns</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>heat.res.row.graph <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>  heat.cluster.,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,         <span class="co"># Do not cluster columns (preserves spatial order of tissues)</span></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutree_rows =</span> number_of_clusters,  <span class="co"># Cut the hierarchical clustering into 8 clusters</span></span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">11</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">20</span>,                    <span class="co"># Set figure width</span></span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellwidth =</span> <span class="dv">50</span>                 <span class="co"># Adjust cell width for clear visualization</span></span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-large" class="level2">
<h2 class="anchored" data-anchor-id="heatmap-large">Heatmap Large</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 1: Load Previously Saved Data</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - These files contain the results from the previous analysis</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - 'regulated.genes_large_control.Rdata' holds the list of significantly regulated genes (padj &lt; 0.05)</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co"># - 'mean.by.tissue_large_control.Rdata' contains mean gene expression values per tissue (VST transformed)</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"regulated.genes_large_control.Rdata"</span>)   <span class="co"># Load significantly regulated genes</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"mean.by.tissue_large_control.Rdata"</span>)    <span class="co"># Load mean expression data for each tissue</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 2: Define Number of Clusters for Heatmap</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="co"># - This determines how the hierarchical clustering will be cut into distinct clusters</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>number_of_clusters <span class="ot">&lt;-</span> <span class="dv">8</span>   <span class="co"># Adjust as needed</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 3: Filter Expression Data for Significantly Regulated Genes</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="co"># - Extract only the genes from 'mean.by.tissue.' that are in the 'regulated_genes.' list</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>heat.cluster. <span class="ot">&lt;-</span> mean.by.tissue.[<span class="fu">rownames</span>(mean.by.tissue.) <span class="sc">%in%</span> regulated_genes., ]</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 4: Generate Heatmap to Visualise Spatial Expression Patterns</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>heat.res.row.graph <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>  heat.cluster.,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,         <span class="co"># Do not cluster columns (preserves spatial order of tissues)</span></span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutree_rows =</span> number_of_clusters,  <span class="co"># Cut the hierarchical clustering into 8 clusters</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">11</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">20</span>,                    <span class="co"># Set figure width</span></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellwidth =</span> <span class="dv">50</span> )                <span class="co"># Adjust cell width for clear visualization</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="healthy-intestines---plot-individual-clusters" class="level2">
<h2 class="anchored" data-anchor-id="healthy-intestines---plot-individual-clusters">Healthy intestines - Plot individual clusters</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Set to "small" or "large"</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>region <span class="ot">&lt;-</span> <span class="st">"small"</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (region <span class="sc">==</span> <span class="st">"large"</span>) {</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">"regulated.genes_large_control.Rdata"</span>)   <span class="co"># Load significantly regulated genes for large intestine</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">"mean.by.tissue_large_control.Rdata"</span>)    <span class="co"># Load mean expression data for large intestine</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span> (region <span class="sc">==</span> <span class="st">"small"</span>) {</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">"regulated.genes_small_control.Rdata"</span>)   <span class="co"># Load significantly regulated genes for small intestine</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">"mean.by.tissue_small_control.Rdata"</span>)    <span class="co"># Load mean expression data for small intestine</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">"Region must be either 'small' or 'large'"</span>)</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Define region-specific settings</span></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (region <span class="sc">==</span> <span class="st">"large"</span>) {</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>  region_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Cecum"</span>, <span class="st">"Ascending Colon"</span>, <span class="st">"Transverse Colon"</span>, </span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"Descending Colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>)</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>  region_labeltext <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Caecum"</span>, <span class="st">"Asc. colon"</span>, <span class="st">"Trans. colon"</span>, </span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Desc. colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>)</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>  output_prefix <span class="ot">&lt;-</span> <span class="st">"Large"</span></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span> (region <span class="sc">==</span> <span class="st">"small"</span>) {</span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>  region_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Duodenum"</span>, <span class="st">"Treitz"</span>, <span class="st">"Small Intestine 3"</span>, </span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"Small Intestine 4"</span>, <span class="st">"Small Intestine 5"</span>, <span class="st">"Small Intestine 6"</span>, </span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"Small Intestine 7"</span>, <span class="st">"Small Intestine 8"</span>, <span class="st">"Small Intestine 9"</span>, </span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"Ileocecal"</span>)</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>  region_labeltext <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Duodenum"</span>, <span class="st">"Treitz"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"Ileocecal"</span>)</span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>  output_prefix <span class="ot">&lt;-</span> <span class="st">"Small"</span></span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">"Region must be either 'small' or 'large'"</span>)</span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 1: Filter Expression Data for Significantly Regulated Genes</span></span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>number_of_clusters <span class="ot">&lt;-</span> <span class="dv">8</span>   <span class="co"># Adjust as needed</span></span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>heat.cluster. <span class="ot">&lt;-</span> mean.by.tissue.[<span class="fu">rownames</span>(mean.by.tissue.) <span class="sc">%in%</span> regulated_genes., ]</span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Heatmap clustering</span></span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>heat.res.row <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a>  heat.cluster., </span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb49-43"><a href="#cb49-43" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb49-44"><a href="#cb49-44" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, </span>
<span id="cb49-45"><a href="#cb49-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span></span>
<span id="cb49-46"><a href="#cb49-46" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Add cluster assignments</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>heat.res.row.clust <span class="ot">&lt;-</span> <span class="fu">cbind</span>(</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  heat.cluster., </span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster =</span> <span class="fu">cutree</span>(heat.res.row<span class="sc">$</span>tree_row, <span class="at">k =</span> number_of_clusters)</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Reorder rows according to dendrogram</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>heat. <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(heat.res.row.clust[heat.res.row<span class="sc">$</span>tree_row<span class="sc">$</span>order, ])</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Relabel clusters sequentially (1 to number_of_clusters)</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>cluster_map <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">former =</span> <span class="fu">unique</span>(heat.<span class="sc">$</span>cluster), <span class="at">new =</span> <span class="fu">seq_len</span>(number_of_clusters))</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>heat. <span class="ot">&lt;-</span>  <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(cluster_map<span class="sc">$</span>former, <span class="cf">function</span>(x) {</span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> heat.[heat.<span class="sc">$</span>cluster <span class="sc">==</span> x, ]</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>  tmp<span class="sc">$</span>cluster <span class="ot">&lt;-</span> cluster_map[cluster_map<span class="sc">$</span>former <span class="sc">==</span> x, ]<span class="sc">$</span>new</span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>})) </span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Summarise genes per cluster</span></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>genes.in.cluster <span class="ot">&lt;-</span> heat. <span class="sc">%&gt;%</span> </span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> </span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Create list of gene IDs per cluster</span></span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a>clusters.num <span class="ot">&lt;-</span> <span class="fu">seq_len</span>(number_of_clusters)</span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a>heat.cluster. <span class="ot">&lt;-</span> <span class="fu">lapply</span>(clusters.num, <span class="cf">function</span>(x) {</span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(<span class="fu">subset</span>(heat., cluster <span class="sc">==</span> x))</span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(heat.cluster.) <span class="ot">&lt;-</span> clusters.num</span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a><span class="do">## Perform GO enrichment per cluster</span></span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a>heat.clust.go. <span class="ot">&lt;-</span> <span class="fu">lapply</span>(heat.cluster., <span class="cf">function</span>(gene_ids) {</span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enrichGO</span>(</span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene     =</span> gene_ids,</span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">OrgDb    =</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">keyType  =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">ont      =</span> <span class="st">"BP"</span></span>
<span id="cb50-38"><a href="#cb50-38" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(heat.clust.go.) <span class="ot">&lt;-</span> clusters.num</span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(genes.in.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    cluster           n         
 Min.   :1.00   Min.   : 134.0  
 1st Qu.:2.75   1st Qu.: 319.2  
 Median :4.50   Median : 910.5  
 Mean   :4.50   Mean   :1060.9  
 3rd Qu.:6.25   3rd Qu.:1594.8  
 Max.   :8.00   Max.   :2588.0  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(heat.clust.go.)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Length Class      Mode
1 12     data.frame list
2 12     data.frame list
3 12     data.frame list
4 12     data.frame list
5 12     data.frame list
6 12     data.frame list
7 12     data.frame list
8 12     data.frame list</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Convert cluster column to factor</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>heat.<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="fu">factor</span>(heat.<span class="sc">$</span>cluster, <span class="at">levels =</span> clusters.num)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot all genes per cluster</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">levels</span>(heat.<span class="sc">$</span>cluster)) {</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset genes for the cluster</span></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>  genes_in_cluster <span class="ot">&lt;-</span> heat. <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cluster <span class="sc">==</span> i)</span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scale gene expression per gene (row-wise z-score)</span></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>  expr_scaled <span class="ot">&lt;-</span> genes_in_cluster <span class="sc">%&gt;%</span></span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(region_columns)) <span class="sc">%&gt;%</span></span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale</span>() <span class="sc">%&gt;%</span></span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>  expr_scaled<span class="sc">$</span>genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(genes_in_cluster)</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate and scale cluster-mean expression</span></span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a>  mean_expr <span class="ot">&lt;-</span> <span class="fu">colMeans</span>(genes_in_cluster <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(region_columns)))</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a>  mean_expr_scaled <span class="ot">&lt;-</span> <span class="fu">scale</span>(mean_expr)</span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a>  mean_expr_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">condition   =</span> <span class="fu">names</span>(mean_expr),</span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">measurement =</span> <span class="fu">as.numeric</span>(mean_expr_scaled)</span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb54-28"><a href="#cb54-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-29"><a href="#cb54-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert to long format for plotting</span></span>
<span id="cb54-30"><a href="#cb54-30" aria-hidden="true" tabindex="-1"></a>  expr_long <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">gather</span>(</span>
<span id="cb54-31"><a href="#cb54-31" aria-hidden="true" tabindex="-1"></a>    expr_scaled, <span class="at">key =</span> <span class="st">"condition"</span>, <span class="at">value =</span> <span class="st">"measurement"</span>, </span>
<span id="cb54-32"><a href="#cb54-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">all_of</span>(region_columns), <span class="at">factor_key =</span> <span class="cn">TRUE</span></span>
<span id="cb54-33"><a href="#cb54-33" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb54-34"><a href="#cb54-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-35"><a href="#cb54-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the plot</span></span>
<span id="cb54-36"><a href="#cb54-36" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb54-37"><a href="#cb54-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="at">data =</span> expr_long, <span class="fu">aes</span>(<span class="at">x =</span> condition, <span class="at">y =</span> measurement, <span class="at">group =</span> genes), </span>
<span id="cb54-38"><a href="#cb54-38" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.05</span>, <span class="at">linewidth =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb54-39"><a href="#cb54-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="at">data =</span> mean_expr_df, <span class="fu">aes</span>(<span class="at">x =</span> condition, <span class="at">y =</span> measurement, <span class="at">group =</span> <span class="dv">1</span>), </span>
<span id="cb54-40"><a href="#cb54-40" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">'#67A9CF'</span>, <span class="at">linewidth =</span> <span class="fl">2.0</span>) <span class="sc">+</span></span>
<span id="cb54-41"><a href="#cb54-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> region_labeltext) <span class="sc">+</span></span>
<span id="cb54-42"><a href="#cb54-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_gut</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span> </span>
<span id="cb54-43"><a href="#cb54-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb54-44"><a href="#cb54-44" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text       =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb54-45"><a href="#cb54-45" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x    =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb54-46"><a href="#cb54-46" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title      =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb54-47"><a href="#cb54-47" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb54-48"><a href="#cb54-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Normalized gene expression"</span>)</span>
<span id="cb54-49"><a href="#cb54-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-50"><a href="#cb54-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Print and save plot</span></span>
<span id="cb54-51"><a href="#cb54-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb54-52"><a href="#cb54-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#graph2ppt(x = p, file = paste0(output_prefix, "_Cluster_", i, ".pptx"), width = 17.42 / 2, height = 8.08 / 2)</span></span>
<span id="cb54-53"><a href="#cb54-53" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-8-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="type-2-diabetes---analysis-of-the-small-intestines" class="level2">
<h2 class="anchored" data-anchor-id="type-2-diabetes---analysis-of-the-small-intestines">Type 2 diabetes - Analysis of the small intestines</h2>
<section id="small-intestines---setup-for-analysis" class="level3">
<h3 class="anchored" data-anchor-id="small-intestines---setup-for-analysis">Small intestines - Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset data for small intestine samples</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> <span class="fu">subset</span>(design_df, <span class="fu">as.numeric</span>(Location.num) <span class="sc">%in%</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>) </span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>count. <span class="ot">&lt;-</span> df[, <span class="fu">rownames</span>(design.)]</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create DESeq2 object and normalize</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a><span class="co"># - This normalizes counts while accounting for differences in sequencing depth</span></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(count., design., <span class="at">design =</span> <span class="sc">~</span>ID)</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>Tissue <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds<span class="sc">$</span>Tissue, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Duodenum"</span>, <span class="st">"Treitz"</span>, <span class="st">"Small intestine 3"</span>, <span class="st">"Small intestine 4"</span>, <span class="st">"Small intestine 5"</span>, <span class="st">"Small intestine 6"</span>))</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>Treatment <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds<span class="sc">$</span>Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T2DM"</span>, <span class="st">"Kontrol"</span>))</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dds) <span class="ot">&lt;-</span> <span class="er">~</span> Tissue <span class="sc">+</span> Treatment</span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression analysis</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">results</span>(dds, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Treatment"</span>, <span class="st">"T2DM"</span>, <span class="st">"Kontrol"</span>))) </span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a><span class="co"># save the DESeq2 object for further analysis</span></span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(dds, <span class="at">file =</span> <span class="st">"forsmallintestineanalysis.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="small-intestines---run-analysis" class="level3">
<h3 class="anchored" data-anchor-id="small-intestines---run-analysis">Small intestines - Run analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"forsmallintestineanalysis.RData"</span>)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert from Ensembl to Entrez</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>converted <span class="ot">&lt;-</span> <span class="fu">bitr</span>(</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(dds),</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">toType   =</span> <span class="fu">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"GENENAME"</span>, <span class="st">"SYMBOL"</span>),</span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">OrgDb    =</span> <span class="st">"org.Hs.eg.db"</span></span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bitr(rownames(dds), fromType = "ENSEMBL", toType = c("ENTREZID", :
2.8% of input gene IDs are fail to map...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge and filter to valid Entrez IDs</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>ensembl_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(dds)</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(converted, dds,</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ENSEMBL"</span> <span class="ot">=</span> <span class="st">"ensembl_id"</span>))</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="sc">!</span><span class="fu">is.na</span>(dds<span class="sc">$</span>ENTREZID), ]</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Store final dataset</span></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>dds_small <span class="ot">&lt;-</span> dds</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a significance cutoff (e.g., padj &lt; 0.05 and log2FoldChange &gt; 1)</span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>res_df_small <span class="ot">&lt;-</span> dds_small <span class="sc">%&gt;%</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">significant =</span> <span class="fu">ifelse</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">"Significant"</span>, <span class="st">"Not significant"</span>))</span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a direction column based on log2FoldChange and padj</span></span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a>res_df_small <span class="ot">&lt;-</span> res_df_small <span class="sc">%&gt;%</span></span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">case_when</span>(</span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>  <span class="sc">~</span> <span class="st">"Up"</span>,</span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">~</span> <span class="st">"Down"</span>,</span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span>                              <span class="sc">~</span> <span class="st">"NS"</span></span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a>  ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="small-intestines---go-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="small-intestines---go-enrichment">Small intestines - GO enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform enrichGO analysis on all significant genes</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>GO_small <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene          =</span> dds_small <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL),</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ont           =</span> <span class="st">"BP"</span>,             <span class="co"># Biological Processes</span></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">keyType       =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">readable      =</span> <span class="cn">TRUE</span></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise the top GO terms</span></span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a>heatmap_GO_small <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(GO_small)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,] <span class="sc">%&gt;%</span> </span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Description)) <span class="sc">%&gt;%</span> </span>
<span id="cb61-15"><a href="#cb61-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">"GO pathways"</span>, <span class="at">y =</span> <span class="fu">reorder</span>(<span class="fu">capitalize_first</span>(Description), <span class="sc">-</span><span class="fu">log10</span>(p.adjust)))) <span class="sc">+</span></span>
<span id="cb61-16"><a href="#cb61-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb61-17"><a href="#cb61-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">high =</span><span class="st">"#41ab5d"</span>, <span class="at">low =</span> <span class="st">"#c7e9c0"</span>) <span class="sc">+</span></span>
<span id="cb61-18"><a href="#cb61-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb61-19"><a href="#cb61-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb61-20"><a href="#cb61-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb61-21"><a href="#cb61-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"-log10(P adj)"</span>,</span>
<span id="cb61-22"><a href="#cb61-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb61-23"><a href="#cb61-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_gut</span>()<span class="sc">+</span></span>
<span id="cb61-24"><a href="#cb61-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb61-25"><a href="#cb61-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-26"><a href="#cb61-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb61-27"><a href="#cb61-27" aria-hidden="true" tabindex="-1"></a>heatmap_GO_small</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = heatmap_GO_small, "heatmap_GO_small", width = 17/3, height = 12/3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="small-intestines---volcano-plot" class="level3">
<h3 class="anchored" data-anchor-id="small-intestines---volcano-plot">Small intestines - Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify top genes for labeling in the volcano plot</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> res_df_small <span class="sc">%&gt;%</span> </span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ENSEMBL, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(direction <span class="sc">!=</span> <span class="st">"NS"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(direction) <span class="sc">%&gt;%</span></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a> <span class="fu">slice_min</span>(<span class="at">order_by =</span> padj, <span class="at">n =</span> <span class="dv">25</span>) <span class="sc">%&gt;%</span>  <span class="co"># top 25 per group</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() </span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the volcano plot</span></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>volcano_plot_small <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res_df_small, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> direction)) <span class="sc">+</span></span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.3</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> direction), <span class="at">size =</span> <span class="fl">2.5</span>) <span class="sc">+</span>  </span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">label =</span> SYMBOL, <span class="at">color =</span> direction), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span> </span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Down"</span> <span class="ot">=</span> <span class="st">"#2166ac"</span>, <span class="st">"Up"</span> <span class="ot">=</span> <span class="st">"#b2182b"</span>, <span class="st">"NS"</span> <span class="ot">=</span> <span class="st">"grey70"</span>)) <span class="sc">+</span></span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"grey50"</span>) <span class="sc">+</span></span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"grey50"</span>) <span class="sc">+</span></span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_gut</span>() <span class="sc">+</span></span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>,</span>
<span id="cb63-21"><a href="#cb63-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"-Log10(P adj)"</span>,</span>
<span id="cb63-22"><a href="#cb63-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Regulation"</span></span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">2.5</span>,<span class="fl">2.5</span>)</span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-27"><a href="#cb63-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the volcano plot</span></span>
<span id="cb63-28"><a href="#cb63-28" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcano_plot_small)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 3740 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the volcano plot to a PowerPoint file</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = volcano_plot_small, "volcano_plot_small", width = 20.37/2.5, height = 11.99/2.5)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="type-2-diabetes---analysis-of-the-distal-small-intestines" class="level2">
<h2 class="anchored" data-anchor-id="type-2-diabetes---analysis-of-the-distal-small-intestines">Type 2 diabetes - Analysis of the distal small intestines</h2>
<section id="distal-small-intestines---setup-for-analysis" class="level3">
<h3 class="anchored" data-anchor-id="distal-small-intestines---setup-for-analysis">Distal small intestines - Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset metadata and count matrix for small intestine samples</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> design_df</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>count. <span class="ot">&lt;-</span> df[, <span class="fu">rownames</span>(design.)]</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DESeq2 dataset</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="co"># - This normalizes counts while accounting for differences in sequencing depth</span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(count., design., <span class="at">design =</span> <span class="sc">~</span>ID)</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>Tissue <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds<span class="sc">$</span>Tissue, <span class="at">levels =</span> tissue_levels_all)</span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>Treatment <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds<span class="sc">$</span>Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T2DM"</span>, <span class="st">"Kontrol"</span>))</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2 analysis</span></span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the DESeq2 object for further analysis</span></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(dds, <span class="at">file =</span> <span class="st">"forsingleregionanalysis.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="distal-small-intestines---run-analysis" class="level3">
<h3 class="anchored" data-anchor-id="distal-small-intestines---run-analysis">Distal small intestines - Run analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"forsingleregionanalysis.RData"</span>)</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results for T2DM vs Kontrol comparison</span></span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">results</span>(dds,<span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ID"</span>, <span class="st">"ileocecal_T2DM"</span>, <span class="st">"ileocecal_Kontrol"</span>)))</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert from Ensembl to Entrez</span></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>converted <span class="ot">&lt;-</span> <span class="fu">bitr</span>(</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(dds),</span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">toType   =</span> <span class="fu">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"GENENAME"</span>, <span class="st">"SYMBOL"</span>),</span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">OrgDb    =</span> <span class="st">"org.Hs.eg.db"</span></span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bitr(rownames(dds), fromType = "ENSEMBL", toType = c("ENTREZID", :
2.8% of input gene IDs are fail to map...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge and filter to valid Entrez IDs</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>ensembl_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(dds)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(converted, dds,</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ENSEMBL"</span> <span class="ot">=</span> <span class="st">"ensembl_id"</span>))</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="sc">!</span><span class="fu">is.na</span>(dds<span class="sc">$</span>ENTREZID), ]</span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Store final dataset</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>dds_ileocecal <span class="ot">&lt;-</span> dds</span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a significance cutoff (e.g., padj &lt; 0.05 and log2FoldChange &gt; 1)</span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>res_df_ileocecal <span class="ot">&lt;-</span> dds_ileocecal <span class="sc">%&gt;%</span></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">significant =</span> <span class="fu">ifelse</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">"Significant"</span>, <span class="st">"Not significant"</span>))</span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a direction column based on log2FoldChange and padj</span></span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>res_df_ileocecal <span class="ot">&lt;-</span> res_df_ileocecal <span class="sc">%&gt;%</span></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">case_when</span>(</span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>  <span class="sc">~</span> <span class="st">"Up"</span>,</span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">~</span> <span class="st">"Down"</span>,</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span>                              <span class="sc">~</span> <span class="st">"NS"</span></span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a>  ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="distal-small-intestines---go-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="distal-small-intestines---go-enrichment">Distal small intestines - GO enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform enrichGO analysis on all significant genes</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>ego_all_ileocecal <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene          =</span> dds_ileocecal <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL),</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ont           =</span> <span class="st">"BP"</span>,             <span class="co"># Biological Processes</span></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">keyType       =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">readable      =</span> <span class="cn">TRUE</span></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Visualise the top GO terms</span></span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>heatmap_GO_ileocecal <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ego_all_ileocecal)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,] <span class="sc">%&gt;%</span> </span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Description)) <span class="sc">%&gt;%</span> </span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">"GO pathways"</span>, <span class="at">y =</span> <span class="fu">reorder</span>(<span class="fu">capitalize_first</span>(Description), <span class="sc">-</span><span class="fu">log10</span>(p.adjust)))) <span class="sc">+</span></span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">high =</span> <span class="st">"#c7e9c0"</span>, <span class="at">low =</span>  <span class="st">"#41ab5d"</span>) <span class="sc">+</span></span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"-log10(P adj)"</span>,</span>
<span id="cb71-23"><a href="#cb71-23" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb71-24"><a href="#cb71-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_gut</span>()<span class="sc">+</span></span>
<span id="cb71-25"><a href="#cb71-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb71-26"><a href="#cb71-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-27"><a href="#cb71-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb71-28"><a href="#cb71-28" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(heatmap_GO_ileocecal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = heatmap_GO_ileocecal, "heatmap_GO_ileocecal", width = 17/3, height = 12/3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="distal-small-intestines---volcano-plot" class="level3">
<h3 class="anchored" data-anchor-id="distal-small-intestines---volcano-plot">Distal small intestines - Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify top genes for labeling in the volcano plot</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> res_df_ileocecal <span class="sc">%&gt;%</span> </span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ENSEMBL, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(direction <span class="sc">!=</span> <span class="st">"NS"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(direction) <span class="sc">%&gt;%</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_min</span>(<span class="at">order_by =</span> padj, <span class="at">n =</span> <span class="dv">25</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() </span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the volcano plot</span></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>volcano_plot_ileocecal <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res_df_ileocecal, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> direction)) <span class="sc">+</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.3</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> direction), <span class="at">size =</span> <span class="fl">2.5</span>) <span class="sc">+</span>  <span class="co"># Larger points</span></span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> top_genes,</span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">label =</span> SYMBOL, <span class="at">color =</span> direction),</span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">max.overlaps =</span> <span class="dv">50</span>,</span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Down"</span> <span class="ot">=</span> <span class="st">"#2166ac"</span>, <span class="st">"Up"</span> <span class="ot">=</span> <span class="st">"#b2182b"</span>, <span class="st">"NS"</span> <span class="ot">=</span> <span class="st">"grey70"</span>)) <span class="sc">+</span></span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"grey50"</span>) <span class="sc">+</span></span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"grey50"</span>) <span class="sc">+</span></span>
<span id="cb73-23"><a href="#cb73-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_gut</span>() <span class="sc">+</span></span>
<span id="cb73-24"><a href="#cb73-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb73-25"><a href="#cb73-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>,</span>
<span id="cb73-26"><a href="#cb73-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"-Log10 Adjusted P-value"</span>,</span>
<span id="cb73-27"><a href="#cb73-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Regulation"</span></span>
<span id="cb73-28"><a href="#cb73-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb73-29"><a href="#cb73-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb73-30"><a href="#cb73-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">20</span>,<span class="dv">20</span>)</span>
<span id="cb73-31"><a href="#cb73-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb73-32"><a href="#cb73-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the volcano plot</span></span>
<span id="cb73-33"><a href="#cb73-33" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcano_plot_ileocecal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 5168 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the volcano plot to a PowerPoint file</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = volcano_plot_ileocecal, "volcano_plot_ileocecal", width = 20.37/2.5, height = 11.99/2.5)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="type-2-diabetes---analysis-of-the-large-intestines" class="level2">
<h2 class="anchored" data-anchor-id="type-2-diabetes---analysis-of-the-large-intestines">Type 2 diabetes - Analysis of the large intestines</h2>
<section id="large-intestines---setup-for-analysis" class="level3">
<h3 class="anchored" data-anchor-id="large-intestines---setup-for-analysis">Large intestines - Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset data for large intestine</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>design. <span class="ot">&lt;-</span> <span class="fu">subset</span>(design_df, <span class="fu">as.numeric</span>(Location.num) <span class="sc">&gt;</span> <span class="dv">10</span>)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>count. <span class="ot">&lt;-</span> df[, <span class="fu">rownames</span>(design.)]</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create DESeq2 object and normalize</span></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>dds. <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(count., design., <span class="at">design =</span> <span class="sc">~</span>ID)</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>ddsTT <span class="ot">&lt;-</span> dds.</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>ddsTT<span class="sc">$</span>Tissue <span class="ot">&lt;-</span> <span class="fu">factor</span>(ddsTT<span class="sc">$</span>Tissue, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Cecum"</span>, <span class="st">"Ascending colon"</span>, <span class="st">"Transverse colon"</span>, <span class="st">"Descending colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>))</span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>ddsTT<span class="sc">$</span>Treatment <span class="ot">&lt;-</span> <span class="fu">factor</span>(ddsTT<span class="sc">$</span>Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T2DM"</span>, <span class="st">"Kontrol"</span>))</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(ddsTT) <span class="ot">&lt;-</span> <span class="er">~</span> Tissue <span class="sc">+</span> Treatment</span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>ddsTT <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ddsTT)</span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression analysis</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a>ddsT <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">results</span>(ddsTT, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"Treatment"</span>, <span class="st">"T2DM"</span>, <span class="st">"Kontrol"</span>))) </span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the DESeq2 object for further analysis</span></span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ddsT, <span class="at">file =</span> <span class="st">"forlargeintestinalanalysis.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="large-intestines---run-analysis" class="level3">
<h3 class="anchored" data-anchor-id="large-intestines---run-analysis">Large intestines - Run analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"forlargeintestinalanalysis.RData"</span>)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert from Ensembl to Entrez</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>converted <span class="ot">&lt;-</span> <span class="fu">bitr</span>(</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(ddsT),</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">toType   =</span> <span class="fu">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"GENENAME"</span>, <span class="st">"SYMBOL"</span>),</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">OrgDb    =</span> <span class="st">"org.Hs.eg.db"</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bitr(rownames(ddsT), fromType = "ENSEMBL", toType = c("ENTREZID", :
2.8% of input gene IDs are fail to map...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge and filter to valid Entrez IDs</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>ddsT<span class="sc">$</span>ensembl_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ddsT)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>ddsT <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(converted, ddsT,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ENSEMBL"</span> <span class="ot">=</span> <span class="st">"ensembl_id"</span>))</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>ddsT <span class="ot">&lt;-</span> ddsT[<span class="sc">!</span><span class="fu">is.na</span>(dds<span class="sc">$</span>ENTREZID), ]</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Store final dataset</span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>ddsT_large <span class="ot">&lt;-</span> ddsT</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a significance cutoff (e.g., padj &lt; 0.05 and log2FoldChange &gt; 1)</span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>res_df_large <span class="ot">&lt;-</span> ddsT_large <span class="sc">%&gt;%</span> </span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">significant =</span> <span class="fu">ifelse</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">"Significant"</span>, <span class="st">"Not significant"</span>))</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a direction column based on log2FoldChange and padj</span></span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>res_df_large <span class="ot">&lt;-</span> res_df_large <span class="sc">%&gt;%</span></span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">case_when</span>(</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>  <span class="sc">~</span> <span class="st">"Up"</span>,</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">~</span> <span class="st">"Down"</span>,</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span>                              <span class="sc">~</span> <span class="st">"NS"</span></span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>  ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="large-intestines---go-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="large-intestines---go-enrichment">Large intestines - GO enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform enrichGO analysis on all significant genes</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>ego_all_large <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene          =</span> (ddsT_large <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(ENSEMBL)),</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ont           =</span> <span class="st">"BP"</span>,             <span class="co"># Biological Processes</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">keyType       =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">readable      =</span> <span class="cn">TRUE</span></span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p.adjust) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise the top GO terms</span></span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>heatmap_GO_large <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ego_all_large)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,] <span class="sc">%&gt;%</span> </span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Description)) <span class="sc">%&gt;%</span> </span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">"GO pathways"</span>, <span class="at">y =</span> <span class="fu">reorder</span>(<span class="fu">capitalize_first</span>(Description), <span class="sc">-</span><span class="fu">log10</span>(p.adjust)))) <span class="sc">+</span></span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb83-18"><a href="#cb83-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">high =</span> <span class="st">"#c7e9c0"</span>, <span class="at">low =</span>  <span class="st">"#41ab5d"</span>) <span class="sc">+</span></span>
<span id="cb83-19"><a href="#cb83-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb83-20"><a href="#cb83-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb83-21"><a href="#cb83-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb83-22"><a href="#cb83-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"-log10(P adj)"</span>,</span>
<span id="cb83-23"><a href="#cb83-23" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb83-24"><a href="#cb83-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_gut</span>()<span class="sc">+</span></span>
<span id="cb83-25"><a href="#cb83-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb83-26"><a href="#cb83-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-27"><a href="#cb83-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb83-28"><a href="#cb83-28" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(heatmap_GO_large)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = heatmap_GO_large, "heatmap_GO_large", width = 17/3, height = 12/3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="large-intestines---volcano-plot" class="level3">
<h3 class="anchored" data-anchor-id="large-intestines---volcano-plot">Large intestines - Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify top genes for labeling in the volcano plot</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> res_df_large <span class="sc">%&gt;%</span> </span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ENSEMBL, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(direction <span class="sc">!=</span> <span class="st">"NS"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(direction) <span class="sc">%&gt;%</span></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_min</span>(<span class="at">order_by =</span> padj, <span class="at">n =</span> <span class="dv">25</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() </span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the volcano plot</span></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>volcano_plot_large <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res_df_large, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> direction)) <span class="sc">+</span></span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.3</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> direction), <span class="at">size =</span> <span class="fl">2.5</span>) <span class="sc">+</span>  <span class="co"># Larger points</span></span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> top_genes,</span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">label =</span> SYMBOL, <span class="at">color =</span> direction),</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">max.overlaps =</span> <span class="dv">50</span></span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Down"</span> <span class="ot">=</span> <span class="st">"#2166ac"</span>, <span class="st">"Up"</span> <span class="ot">=</span> <span class="st">"#b2182b"</span>, <span class="st">"NS"</span> <span class="ot">=</span> <span class="st">"grey70"</span>)) <span class="sc">+</span></span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"grey50"</span>) <span class="sc">+</span></span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"grey50"</span>) <span class="sc">+</span></span>
<span id="cb85-23"><a href="#cb85-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_gut</span>() <span class="sc">+</span></span>
<span id="cb85-24"><a href="#cb85-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb85-25"><a href="#cb85-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>,</span>
<span id="cb85-26"><a href="#cb85-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"-Log10 Adjusted P-value"</span>,</span>
<span id="cb85-27"><a href="#cb85-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Regulation"</span></span>
<span id="cb85-28"><a href="#cb85-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb85-29"><a href="#cb85-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb85-30"><a href="#cb85-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">3</span>)</span>
<span id="cb85-31"><a href="#cb85-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-32"><a href="#cb85-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the volcano plot</span></span>
<span id="cb85-33"><a href="#cb85-33" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcano_plot_large)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 3361 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the volcano plot to a PowerPoint file</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = volcano_plot_large, "volcano_plot_large", width = 20.37/2.5, height = 11.99/2.5)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rpkm-plot-of-individual-genes-across-16-anatomical-regions-of-the-intestine" class="level3">
<h3 class="anchored" data-anchor-id="rpkm-plot-of-individual-genes-across-16-anatomical-regions-of-the-intestine">RPKM plot of individual genes across 16 anatomical regions of the intestine</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Plot gene expression for individual genes across all regions</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>plot_gene_expression <span class="ot">&lt;-</span> <span class="cf">function</span>(expr_df, gene_id, </span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">gene_info_small =</span> res_df_small,</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">gene_info_ileocecal =</span> res_df_ileocecal,</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">gene_info_large =</span> res_df_large) {</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Region labels for x-axis</span></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>  labeltext <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'Duodenum'</span>, <span class="st">'Treitz'</span>,<span class="st">'3'</span>,<span class="st">'4'</span>,<span class="st">'5'</span>,<span class="st">'6'</span>,<span class="st">'7'</span>,<span class="st">'8'</span>,<span class="st">'9'</span>,<span class="st">'Ileocaecal'</span>,<span class="st">"Caecum"</span>, <span class="st">"Asc. colon"</span>, <span class="st">"Trans. colon"</span>, <span class="st">"Desc. colon"</span>, <span class="st">"Sigmoideum"</span>, <span class="st">"Rectum"</span>)</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 1. Filter RPKM data for selected gene</span></span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>  plot_df <span class="ot">&lt;-</span> expr_df <span class="sc">%&gt;%</span></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ENSEMBL <span class="sc">==</span> gene_id) </span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 2. Helper function: Format direction + padj or NS</span></span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a>format_stats <span class="ot">&lt;-</span> <span class="cf">function</span>(df, region_name) {</span>
<span id="cb90-15"><a href="#cb90-15" aria-hidden="true" tabindex="-1"></a>  row <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ENSEMBL <span class="sc">==</span> gene_id)</span>
<span id="cb90-16"><a href="#cb90-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(row) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">||</span> <span class="fu">any</span>(<span class="fu">is.na</span>(row<span class="sc">$</span>padj))) <span class="fu">return</span>(<span class="fu">paste0</span>(region_name, <span class="st">": NS"</span>))</span>
<span id="cb90-17"><a href="#cb90-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-18"><a href="#cb90-18" aria-hidden="true" tabindex="-1"></a>  row <span class="ot">&lt;-</span> row <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="dv">1</span>)  <span class="co"># pick the "most significant" one if duplicates</span></span>
<span id="cb90-19"><a href="#cb90-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-20"><a href="#cb90-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (row<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb90-21"><a href="#cb90-21" aria-hidden="true" tabindex="-1"></a>    arrow <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(row<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"↑"</span>, <span class="st">"↓"</span>)</span>
<span id="cb90-22"><a href="#cb90-22" aria-hidden="true" tabindex="-1"></a>    padj_label <span class="ot">&lt;-</span> <span class="fu">formatC</span>(row<span class="sc">$</span>padj, <span class="at">format =</span> <span class="st">"e"</span>, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb90-23"><a href="#cb90-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">paste0</span>(region_name, <span class="st">":"</span>, arrow, <span class="st">"padj = ("</span>, padj_label, <span class="st">")"</span>))</span>
<span id="cb90-24"><a href="#cb90-24" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb90-25"><a href="#cb90-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">paste0</span>(region_name, <span class="st">": NS"</span>))</span>
<span id="cb90-26"><a href="#cb90-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb90-27"><a href="#cb90-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb90-28"><a href="#cb90-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-29"><a href="#cb90-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 3. Build subtitle with all 3 regions</span></span>
<span id="cb90-30"><a href="#cb90-30" aria-hidden="true" tabindex="-1"></a>  subtitle_text <span class="ot">&lt;-</span> <span class="fu">paste</span>(</span>
<span id="cb90-31"><a href="#cb90-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">format_stats</span>(gene_info_small, <span class="st">"Small intestine"</span>),</span>
<span id="cb90-32"><a href="#cb90-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">format_stats</span>(gene_info_ileocecal, <span class="st">"Ileocaecal"</span>),</span>
<span id="cb90-33"><a href="#cb90-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">format_stats</span>(gene_info_large, <span class="st">"Large intestine"</span>),</span>
<span id="cb90-34"><a href="#cb90-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">" "</span></span>
<span id="cb90-35"><a href="#cb90-35" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb90-36"><a href="#cb90-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-37"><a href="#cb90-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 4. Get gene name and description</span></span>
<span id="cb90-38"><a href="#cb90-38" aria-hidden="true" tabindex="-1"></a>  symbol <span class="ot">&lt;-</span> <span class="fu">unique</span>(plot_df<span class="sc">$</span>SYMBOL)</span>
<span id="cb90-39"><a href="#cb90-39" aria-hidden="true" tabindex="-1"></a>  description <span class="ot">&lt;-</span> <span class="fu">unique</span>(plot_df<span class="sc">$</span>GENENAME)</span>
<span id="cb90-40"><a href="#cb90-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-41"><a href="#cb90-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 5. Compute group-wise mean expression</span></span>
<span id="cb90-42"><a href="#cb90-42" aria-hidden="true" tabindex="-1"></a>  line_df <span class="ot">&lt;-</span> plot_df <span class="sc">%&gt;%</span></span>
<span id="cb90-43"><a href="#cb90-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Type, Location) <span class="sc">%&gt;%</span></span>
<span id="cb90-44"><a href="#cb90-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean_expression =</span> <span class="fu">mean</span>(Expression, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb90-45"><a href="#cb90-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-46"><a href="#cb90-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 6. Plot</span></span>
<span id="cb90-47"><a href="#cb90-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(plot_df, <span class="fu">aes</span>(<span class="at">x =</span> Location, <span class="at">y =</span> Expression)) <span class="sc">+</span></span>
<span id="cb90-48"><a href="#cb90-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"rect"</span>, <span class="at">xmin =</span> <span class="fl">10.5</span>, <span class="at">xmax =</span> <span class="fl">16.7</span>, <span class="at">ymin =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">ymax =</span> <span class="cn">Inf</span>, <span class="at">fill =</span> <span class="st">"grey"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb90-49"><a href="#cb90-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Type), </span>
<span id="cb90-50"><a href="#cb90-50" aria-hidden="true" tabindex="-1"></a>               <span class="at">position =</span> <span class="fu">position_jitterdodge</span>(<span class="at">jitter.width =</span> <span class="fl">0.2</span>, <span class="at">dodge.width =</span> <span class="fl">0.6</span>),</span>
<span id="cb90-51"><a href="#cb90-51" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.3</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb90-52"><a href="#cb90-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="at">data =</span> line_df, <span class="fu">aes</span>(<span class="at">y =</span> mean_expression, <span class="at">group =</span> Type, <span class="at">color =</span> Type), <span class="at">size =</span> <span class="fl">1.3</span>) <span class="sc">+</span></span>
<span id="cb90-53"><a href="#cb90-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Kontrol"</span> <span class="ot">=</span> <span class="st">"grey33"</span>, <span class="st">"T2DM"</span> <span class="ot">=</span> <span class="st">"#4292c6"</span>),   </span>
<span id="cb90-54"><a href="#cb90-54" aria-hidden="true" tabindex="-1"></a>                       <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Kontrol"</span> <span class="ot">=</span> <span class="st">"Control"</span>, <span class="st">"T2DM"</span> <span class="ot">=</span> <span class="st">"T2D"</span>)</span>
<span id="cb90-55"><a href="#cb90-55" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb90-56"><a href="#cb90-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_GE</span>() <span class="sc">+</span></span>
<span id="cb90-57"><a href="#cb90-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb90-58"><a href="#cb90-58" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb90-59"><a href="#cb90-59" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb90-60"><a href="#cb90-60" aria-hidden="true" tabindex="-1"></a>          <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),  <span class="co"># smaller subtitle</span></span>
<span id="cb90-61"><a href="#cb90-61" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">17</span>, .<span class="dv">95</span>),</span>
<span id="cb90-62"><a href="#cb90-62" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="st">"right"</span>, <span class="st">"top"</span>),</span>
<span id="cb90-63"><a href="#cb90-63" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.box.just =</span> <span class="st">"right"</span>,</span>
<span id="cb90-64"><a href="#cb90-64" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb90-65"><a href="#cb90-65" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb90-66"><a href="#cb90-66" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.margin =</span> <span class="fu">margin</span>(<span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb90-67"><a href="#cb90-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb90-68"><a href="#cb90-68" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fu">paste0</span>(symbol, <span class="st">" expression (RPKM)"</span>),</span>
<span id="cb90-69"><a href="#cb90-69" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="fu">paste0</span>(<span class="fu">capitalize_first</span>(description), <span class="st">" ("</span>, symbol,<span class="st">")"</span>),</span>
<span id="cb90-70"><a href="#cb90-70" aria-hidden="true" tabindex="-1"></a>      <span class="at">subtitle =</span> subtitle_text</span>
<span id="cb90-71"><a href="#cb90-71" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb90-72"><a href="#cb90-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> labeltext)</span>
<span id="cb90-73"><a href="#cb90-73" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb90-74"><a href="#cb90-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-75"><a href="#cb90-75" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose genes to plot</span></span>
<span id="cb90-76"><a href="#cb90-76" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(<span class="st">"ENSG00000160221"</span>, <span class="st">"ENSG00000124701"</span>, <span class="st">"ENSG00000179344"</span> )) <span class="co">#Genes small intestine</span></span>
<span id="cb90-77"><a href="#cb90-77" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(<span class="st">"ENSG00000148346"</span>, <span class="st">"ENSG00000166816"</span>, <span class="st">"ENSG00000272398"</span> )) <span class="co">#Genes ileocecal </span></span>
<span id="cb90-78"><a href="#cb90-78" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(<span class="st">"ENSG00000156284"</span>, <span class="st">"ENSG00000050555"</span>, <span class="st">"ENSG00000117148"</span> )) <span class="co">#Genes large </span></span>
<span id="cb90-79"><a href="#cb90-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-80"><a href="#cb90-80" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Loop through each gene and save plots individually</span></span>
<span id="cb90-81"><a href="#cb90-81" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene_id <span class="cf">in</span> gene_names) {</span>
<span id="cb90-82"><a href="#cb90-82" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">plot_gene_expression</span>(rpkm_annotated, gene_id)</span>
<span id="cb90-83"><a href="#cb90-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb90-84"><a href="#cb90-84" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot to a PowerPoint file</span></span>
<span id="cb90-85"><a href="#cb90-85" aria-hidden="true" tabindex="-1"></a>  <span class="co">#graph2ppt(x = plot, paste0(gene_id, "_expression.pptx"), width = 8.887, height = 4.9)</span></span>
<span id="cb90-86"><a href="#cb90-86" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: A numeric `legend.position` argument in `theme()` was deprecated in ggplot2
3.5.0.
ℹ Please use the `legend.position.inside` argument of `theme()` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-21-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-21-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="single-cell-analysis" class="level1">
<h1>Single cell analysis</h1>
<section id="load-everything-in" class="level3">
<h3 class="anchored" data-anchor-id="load-everything-in">Load everything in</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>umap_info_small <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">'/Users/hannahgilliam-vigh/Desktop/Single cell files/50851431_UMAP_info.txt'</span>)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>umap_info_large <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">'/Users/hannahgilliam-vigh/Desktop/Single cell files/50851005_UMAP_info.txt'</span>)</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>gene2acc<span class="ot">=</span><span class="fu">read.delim</span>(<span class="st">'/Users/hannahgilliam-vigh/Desktop/Single cell files/Gene2Ensembl.txt'</span>) <span class="co">#This one translates gene names to ensembl IDs</span></span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename column in gene2acc to match for join</span></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>gene2acc <span class="ot">&lt;-</span> gene2acc <span class="sc">%&gt;%</span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">ENSEMBL =</span> ensembl_gene_id)</span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Library/Mobile Documents/com~apple~CloudDocs/pca_project/R/res_df_small.RData"</span>)</span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Library/Mobile Documents/com~apple~CloudDocs/pca_project/R/res_df_ileocecal.RData"</span>)</span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Library/Mobile Documents/com~apple~CloudDocs/pca_project/R/res_df_large.RData"</span>)</span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a>ranked_cells_large <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Stromal</span></span>
<span id="cb93-17"><a href="#cb93-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"fibroblast"</span>,</span>
<span id="cb93-18"><a href="#cb93-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">"myofibroblast cell"</span>,</span>
<span id="cb93-19"><a href="#cb93-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">"colon endothelial cell"</span>,</span>
<span id="cb93-20"><a href="#cb93-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enteroglial cell"</span>,</span>
<span id="cb93-21"><a href="#cb93-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-22"><a href="#cb93-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Epithelial</span></span>
<span id="cb93-23"><a href="#cb93-23" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enterocyte of epithelium of large intestine"</span>,</span>
<span id="cb93-24"><a href="#cb93-24" aria-hidden="true" tabindex="-1"></a>  <span class="st">"large intestine goblet cell"</span>,</span>
<span id="cb93-25"><a href="#cb93-25" aria-hidden="true" tabindex="-1"></a>  <span class="st">"paneth cell of colon"</span>,</span>
<span id="cb93-26"><a href="#cb93-26" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tuft cell of colon"</span>,</span>
<span id="cb93-27"><a href="#cb93-27" aria-hidden="true" tabindex="-1"></a>  <span class="st">"type l enteroendocrine cell"</span>,</span>
<span id="cb93-28"><a href="#cb93-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enterochromaffin-like cell"</span>,</span>
<span id="cb93-29"><a href="#cb93-29" aria-hidden="true" tabindex="-1"></a>  <span class="st">"transit amplifying cell of colon"</span>,</span>
<span id="cb93-30"><a href="#cb93-30" aria-hidden="true" tabindex="-1"></a>  <span class="st">"intestinal crypt stem cell of colon"</span>,</span>
<span id="cb93-31"><a href="#cb93-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">"best4+ intestinal epithelial cell, human"</span>,</span>
<span id="cb93-32"><a href="#cb93-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-33"><a href="#cb93-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Adaptive immune</span></span>
<span id="cb93-34"><a href="#cb93-34" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cd4-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-35"><a href="#cb93-35" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cd8-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-36"><a href="#cb93-36" aria-hidden="true" tabindex="-1"></a>  <span class="st">"naive thymus-derived cd4-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-37"><a href="#cb93-37" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gamma-delta t cell"</span>,</span>
<span id="cb93-38"><a href="#cb93-38" aria-hidden="true" tabindex="-1"></a>  <span class="st">"regulatory t cell"</span>,</span>
<span id="cb93-39"><a href="#cb93-39" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mature nk t cell"</span>,</span>
<span id="cb93-40"><a href="#cb93-40" aria-hidden="true" tabindex="-1"></a>  <span class="st">"b cell"</span>,</span>
<span id="cb93-41"><a href="#cb93-41" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma cell"</span>,</span>
<span id="cb93-42"><a href="#cb93-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-43"><a href="#cb93-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Innate immune</span></span>
<span id="cb93-44"><a href="#cb93-44" aria-hidden="true" tabindex="-1"></a>  <span class="st">"natural killer cell"</span>,</span>
<span id="cb93-45"><a href="#cb93-45" aria-hidden="true" tabindex="-1"></a>  <span class="st">"innate lymphoid cell"</span>,</span>
<span id="cb93-46"><a href="#cb93-46" aria-hidden="true" tabindex="-1"></a>  <span class="st">"monocyte"</span>,</span>
<span id="cb93-47"><a href="#cb93-47" aria-hidden="true" tabindex="-1"></a>  <span class="st">"colon macrophage"</span>,</span>
<span id="cb93-48"><a href="#cb93-48" aria-hidden="true" tabindex="-1"></a>  <span class="st">"myeloid dendritic cell"</span>,</span>
<span id="cb93-49"><a href="#cb93-49" aria-hidden="true" tabindex="-1"></a>  <span class="st">"neutrophil"</span>,</span>
<span id="cb93-50"><a href="#cb93-50" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mast cell"</span>,</span>
<span id="cb93-51"><a href="#cb93-51" aria-hidden="true" tabindex="-1"></a>  <span class="st">"myeloid leukocyte"</span></span>
<span id="cb93-52"><a href="#cb93-52" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb93-53"><a href="#cb93-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-54"><a href="#cb93-54" aria-hidden="true" tabindex="-1"></a>gaps_col_large <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-55"><a href="#cb93-55" aria-hidden="true" tabindex="-1"></a>  <span class="dv">4</span>,                     <span class="co"># After stromal</span></span>
<span id="cb93-56"><a href="#cb93-56" aria-hidden="true" tabindex="-1"></a>  <span class="dv">13</span>,                    <span class="co"># After epithelial</span></span>
<span id="cb93-57"><a href="#cb93-57" aria-hidden="true" tabindex="-1"></a>  <span class="dv">21</span>                     <span class="co"># After adaptive immune</span></span>
<span id="cb93-58"><a href="#cb93-58" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb93-59"><a href="#cb93-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-60"><a href="#cb93-60" aria-hidden="true" tabindex="-1"></a>epithelial_cells <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-61"><a href="#cb93-61" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enterocyte of epithelium proper of duodenum"</span>,</span>
<span id="cb93-62"><a href="#cb93-62" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enterocyte of epithelium proper of jejunum"</span>,</span>
<span id="cb93-63"><a href="#cb93-63" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enterocyte of epithelium proper of ileum"</span>,</span>
<span id="cb93-64"><a href="#cb93-64" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enterocyte of epithelium proper of small intestine"</span>,</span>
<span id="cb93-65"><a href="#cb93-65" aria-hidden="true" tabindex="-1"></a>  <span class="st">"small intestine goblet cell"</span>,</span>
<span id="cb93-66"><a href="#cb93-66" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enteroendocrine cell of small intestine"</span>,</span>
<span id="cb93-67"><a href="#cb93-67" aria-hidden="true" tabindex="-1"></a>  <span class="st">"intestinal tuft cell"</span>,</span>
<span id="cb93-68"><a href="#cb93-68" aria-hidden="true" tabindex="-1"></a>  <span class="st">"paneth cell of epithelium of small intestine"</span>,</span>
<span id="cb93-69"><a href="#cb93-69" aria-hidden="true" tabindex="-1"></a>  <span class="st">"transit amplifying cell of small intestine"</span>,</span>
<span id="cb93-70"><a href="#cb93-70" aria-hidden="true" tabindex="-1"></a>  <span class="st">"intestinal crypt stem cell of small intestine"</span>,</span>
<span id="cb93-71"><a href="#cb93-71" aria-hidden="true" tabindex="-1"></a>  <span class="st">"best4+ intestinal epithelial cell, human"</span></span>
<span id="cb93-72"><a href="#cb93-72" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb93-73"><a href="#cb93-73" aria-hidden="true" tabindex="-1"></a>stromal_cells <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-74"><a href="#cb93-74" aria-hidden="true" tabindex="-1"></a>  <span class="st">"fibroblast"</span>,</span>
<span id="cb93-75"><a href="#cb93-75" aria-hidden="true" tabindex="-1"></a>  <span class="st">"myofibroblast cell"</span>,</span>
<span id="cb93-76"><a href="#cb93-76" aria-hidden="true" tabindex="-1"></a>  <span class="st">"enteroglial cell"</span>,</span>
<span id="cb93-77"><a href="#cb93-77" aria-hidden="true" tabindex="-1"></a>  <span class="st">"pericyte"</span>,</span>
<span id="cb93-78"><a href="#cb93-78" aria-hidden="true" tabindex="-1"></a>  <span class="st">"endothelial cell of arteriole"</span>,</span>
<span id="cb93-79"><a href="#cb93-79" aria-hidden="true" tabindex="-1"></a>  <span class="st">"endothelial cell of venule"</span>,</span>
<span id="cb93-80"><a href="#cb93-80" aria-hidden="true" tabindex="-1"></a>  <span class="st">"endothelial cell of lymphatic vessel"</span></span>
<span id="cb93-81"><a href="#cb93-81" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb93-82"><a href="#cb93-82" aria-hidden="true" tabindex="-1"></a>adaptive_cells <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-83"><a href="#cb93-83" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cd4-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-84"><a href="#cb93-84" aria-hidden="true" tabindex="-1"></a>  <span class="st">"activated cd4-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-85"><a href="#cb93-85" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cd8-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-86"><a href="#cb93-86" aria-hidden="true" tabindex="-1"></a>  <span class="st">"activated cd8-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-87"><a href="#cb93-87" aria-hidden="true" tabindex="-1"></a>  <span class="st">"naive thymus-derived cd4-positive, alpha-beta t cell"</span>,</span>
<span id="cb93-88"><a href="#cb93-88" aria-hidden="true" tabindex="-1"></a>  <span class="st">"regulatory t cell"</span>,</span>
<span id="cb93-89"><a href="#cb93-89" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gamma-delta t cell"</span>,</span>
<span id="cb93-90"><a href="#cb93-90" aria-hidden="true" tabindex="-1"></a>  <span class="st">"b cell"</span>,</span>
<span id="cb93-91"><a href="#cb93-91" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma cell"</span></span>
<span id="cb93-92"><a href="#cb93-92" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb93-93"><a href="#cb93-93" aria-hidden="true" tabindex="-1"></a>innate_cells <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-94"><a href="#cb93-94" aria-hidden="true" tabindex="-1"></a>  <span class="st">"natural killer cell"</span>,</span>
<span id="cb93-95"><a href="#cb93-95" aria-hidden="true" tabindex="-1"></a>  <span class="st">"innate lymphoid cell"</span>,</span>
<span id="cb93-96"><a href="#cb93-96" aria-hidden="true" tabindex="-1"></a>  <span class="st">"monocyte"</span>,</span>
<span id="cb93-97"><a href="#cb93-97" aria-hidden="true" tabindex="-1"></a>  <span class="st">"macrophage"</span>,</span>
<span id="cb93-98"><a href="#cb93-98" aria-hidden="true" tabindex="-1"></a>  <span class="st">"myeloid dendritic cell"</span>,</span>
<span id="cb93-99"><a href="#cb93-99" aria-hidden="true" tabindex="-1"></a>  <span class="st">"neutrophil"</span>,</span>
<span id="cb93-100"><a href="#cb93-100" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mast cell"</span></span>
<span id="cb93-101"><a href="#cb93-101" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb93-102"><a href="#cb93-102" aria-hidden="true" tabindex="-1"></a>ranked_cells_small <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-103"><a href="#cb93-103" aria-hidden="true" tabindex="-1"></a>  stromal_cells,</span>
<span id="cb93-104"><a href="#cb93-104" aria-hidden="true" tabindex="-1"></a>  epithelial_cells,</span>
<span id="cb93-105"><a href="#cb93-105" aria-hidden="true" tabindex="-1"></a>  adaptive_cells,</span>
<span id="cb93-106"><a href="#cb93-106" aria-hidden="true" tabindex="-1"></a>  innate_cells</span>
<span id="cb93-107"><a href="#cb93-107" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb93-108"><a href="#cb93-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-109"><a href="#cb93-109" aria-hidden="true" tabindex="-1"></a>gaps_col_small <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb93-110"><a href="#cb93-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">length</span>(stromal_cells),                                <span class="co"># after stromal</span></span>
<span id="cb93-111"><a href="#cb93-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">length</span>(stromal_cells) <span class="sc">+</span> <span class="fu">length</span>(epithelial_cells),     <span class="co"># after epithelial</span></span>
<span id="cb93-112"><a href="#cb93-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">length</span>(stromal_cells) <span class="sc">+</span> <span class="fu">length</span>(epithelial_cells) <span class="sc">+</span> <span class="fu">length</span>(adaptive_cells)  <span class="co"># after adaptive</span></span>
<span id="cb93-113"><a href="#cb93-113" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-umaps" class="level3">
<h3 class="anchored" data-anchor-id="plotting-umaps">Plotting UMAPs</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="do">###Plot a UMAP_small</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Coloured by celltype annotation</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_info_small, <span class="fu">aes</span>(Xumap_1,Xumap_2, <span class="at">fill=</span>ident))<span class="sc">+</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">pch=</span><span class="dv">21</span>)<span class="sc">+</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="do">###Plot a UMAP large</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Coloured by celltype annotation</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_info_large, <span class="fu">aes</span>(Xumap_1,Xumap_2, <span class="at">fill=</span>ident))<span class="sc">+</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">pch=</span><span class="dv">21</span>)<span class="sc">+</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-23-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="setup-for-analysis" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>expression_small<span class="ot">=</span><span class="fu">read.delim</span>(<span class="st">'/Users/hannahgilliam-vigh/Desktop/Single cell files/50851431_Normalized_values.txt'</span>)</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>expression_large<span class="ot">=</span><span class="fu">read.delim</span>(<span class="st">'/Users/hannahgilliam-vigh/Desktop/Single cell files/50851005_Normalized_values.txt'</span>)</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(expression_small)</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(expression_large)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="plot-heatmap-of-cluster-in-the-healthy-small-intestine" class="level1">
<h1>Plot heatmap of cluster in the healthy small intestine</h1>
<section id="small-intestine-cluster-2---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="small-intestine-cluster-2---plot-heatmap">Small intestine cluster 2 - Plot heatmap</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Library/Mobile Documents/com~apple~CloudDocs/pca_project/R/heat.cluster.small.Rdata"</span>)</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Library/Mobile Documents/com~apple~CloudDocs/pca_project/R/heat.clust.go_small_control.Rdata"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="setup-for-analysis-1" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-1">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_small</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression dataframe of genes of interest</span></span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]</span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>) <span class="co">#Reformatting</span></span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the cell ID as a column if it's in the rownames</span></span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb98-21"><a href="#cb98-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-22"><a href="#cb98-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge</span></span>
<span id="cb98-23"><a href="#cb98-23" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb98-24"><a href="#cb98-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-25"><a href="#cb98-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb98-26"><a href="#cb98-26" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_small_intestine_cluster2.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-2" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-2">Plot heatmap of cluster 2</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_small_intestine_cluster2.Rdata"</span>)</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by ident and gene, and calculate the mean value for each gene in each cell type</span></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the results into a single data frame</span></span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Remove all genes which have an average of 0 in all cells</span></span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance</span></span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb99-35"><a href="#cb99-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-36"><a href="#cb99-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-37"><a href="#cb99-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb99-38"><a href="#cb99-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-39"><a href="#cb99-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-40"><a href="#cb99-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-41"><a href="#cb99-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb99-42"><a href="#cb99-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb99-43"><a href="#cb99-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-44"><a href="#cb99-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-45"><a href="#cb99-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-46"><a href="#cb99-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb99-47"><a href="#cb99-47" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb99-48"><a href="#cb99-48" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-49"><a href="#cb99-49" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb99-50"><a href="#cb99-50" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb99-51"><a href="#cb99-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-52"><a href="#cb99-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb99-53"><a href="#cb99-53" aria-hidden="true" tabindex="-1"></a> small_intestine_cluster2_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb99-54"><a href="#cb99-54" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb99-55"><a href="#cb99-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb99-56"><a href="#cb99-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb99-57"><a href="#cb99-57" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb99-58"><a href="#cb99-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb99-59"><a href="#cb99-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb99-60"><a href="#cb99-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb99-61"><a href="#cb99-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb99-62"><a href="#cb99-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb99-63"><a href="#cb99-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb99-64"><a href="#cb99-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb99-65"><a href="#cb99-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb99-66"><a href="#cb99-66" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb99-67"><a href="#cb99-67" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb99-68"><a href="#cb99-68" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb99-69"><a href="#cb99-69" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb99-70"><a href="#cb99-70" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb99-71"><a href="#cb99-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-72"><a href="#cb99-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb99-73"><a href="#cb99-73" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(small_intestine_cluster2_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = small_intestine_cluster2_SC, "small_intestine_cluster2_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="small-intestine-cluster-3---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="small-intestine-cluster-3---plot-heatmap">Small intestine cluster 3 - Plot heatmap</h2>
<section id="setup-for-analysis-2" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-2">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_small</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-16"><a href="#cb101-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-17"><a href="#cb101-17" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb101-18"><a href="#cb101-18" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb101-19"><a href="#cb101-19" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb101-20"><a href="#cb101-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-21"><a href="#cb101-21" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb101-22"><a href="#cb101-22" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb101-23"><a href="#cb101-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-24"><a href="#cb101-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb101-25"><a href="#cb101-25" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb101-26"><a href="#cb101-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-27"><a href="#cb101-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb101-28"><a href="#cb101-28" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_small_intestine_cluster3.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-3" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-3">Plot heatmap of cluster 3</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_small_intestine_cluster3.Rdata"</span>)</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-16"><a href="#cb102-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb102-17"><a href="#cb102-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb102-18"><a href="#cb102-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb102-19"><a href="#cb102-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb102-20"><a href="#cb102-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb102-21"><a href="#cb102-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb102-22"><a href="#cb102-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb102-23"><a href="#cb102-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-24"><a href="#cb102-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb102-25"><a href="#cb102-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb102-26"><a href="#cb102-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-27"><a href="#cb102-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb102-28"><a href="#cb102-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb102-29"><a href="#cb102-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-30"><a href="#cb102-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb102-31"><a href="#cb102-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb102-32"><a href="#cb102-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-33"><a href="#cb102-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-34"><a href="#cb102-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb102-35"><a href="#cb102-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-36"><a href="#cb102-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-37"><a href="#cb102-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-38"><a href="#cb102-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb102-39"><a href="#cb102-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb102-40"><a href="#cb102-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-41"><a href="#cb102-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-42"><a href="#cb102-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-43"><a href="#cb102-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb102-44"><a href="#cb102-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb102-45"><a href="#cb102-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-46"><a href="#cb102-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb102-47"><a href="#cb102-47" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb102-48"><a href="#cb102-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-49"><a href="#cb102-49" aria-hidden="true" tabindex="-1"></a>small_intestine_cluster3_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb102-50"><a href="#cb102-50" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb102-51"><a href="#cb102-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb102-52"><a href="#cb102-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb102-53"><a href="#cb102-53" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb102-54"><a href="#cb102-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb102-55"><a href="#cb102-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb102-56"><a href="#cb102-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb102-57"><a href="#cb102-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb102-58"><a href="#cb102-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb102-59"><a href="#cb102-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb102-60"><a href="#cb102-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb102-61"><a href="#cb102-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb102-62"><a href="#cb102-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb102-63"><a href="#cb102-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb102-64"><a href="#cb102-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb102-65"><a href="#cb102-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb102-66"><a href="#cb102-66" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb102-67"><a href="#cb102-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-68"><a href="#cb102-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb102-69"><a href="#cb102-69" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(small_intestine_cluster3_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = small_intestine_cluster3_SC, "small_intestine_cluster3_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="small-intestine-cluster-4---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="small-intestine-cluster-4---plot-heatmap">Small intestine cluster 4 - Plot heatmap</h2>
<section id="setup-for-analysis-3" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-3">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_small</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">4</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">4</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-18"><a href="#cb104-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb104-19"><a href="#cb104-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb104-20"><a href="#cb104-20" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb104-21"><a href="#cb104-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-22"><a href="#cb104-22" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb104-23"><a href="#cb104-23" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb104-24"><a href="#cb104-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-25"><a href="#cb104-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb104-26"><a href="#cb104-26" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb104-27"><a href="#cb104-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-28"><a href="#cb104-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb104-29"><a href="#cb104-29" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_small_intestine_cluster4.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-4" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-4">Plot heatmap of cluster 4</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_small_intestine_cluster4.Rdata"</span>)</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb105-13"><a href="#cb105-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb105-14"><a href="#cb105-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb105-15"><a href="#cb105-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-16"><a href="#cb105-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb105-17"><a href="#cb105-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb105-18"><a href="#cb105-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb105-19"><a href="#cb105-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb105-20"><a href="#cb105-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb105-21"><a href="#cb105-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb105-22"><a href="#cb105-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb105-23"><a href="#cb105-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb105-24"><a href="#cb105-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-25"><a href="#cb105-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb105-26"><a href="#cb105-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-27"><a href="#cb105-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb105-28"><a href="#cb105-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb105-29"><a href="#cb105-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-30"><a href="#cb105-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb105-31"><a href="#cb105-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb105-32"><a href="#cb105-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-33"><a href="#cb105-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb105-34"><a href="#cb105-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-35"><a href="#cb105-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-36"><a href="#cb105-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-37"><a href="#cb105-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb105-38"><a href="#cb105-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb105-39"><a href="#cb105-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-40"><a href="#cb105-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-41"><a href="#cb105-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-42"><a href="#cb105-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb105-43"><a href="#cb105-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb105-44"><a href="#cb105-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-45"><a href="#cb105-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb105-46"><a href="#cb105-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb105-47"><a href="#cb105-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-48"><a href="#cb105-48" aria-hidden="true" tabindex="-1"></a>small_intestine_cluster4_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb105-49"><a href="#cb105-49" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb105-50"><a href="#cb105-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb105-51"><a href="#cb105-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb105-52"><a href="#cb105-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb105-53"><a href="#cb105-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb105-54"><a href="#cb105-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb105-55"><a href="#cb105-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb105-56"><a href="#cb105-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb105-57"><a href="#cb105-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb105-58"><a href="#cb105-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb105-59"><a href="#cb105-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb105-60"><a href="#cb105-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb105-61"><a href="#cb105-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb105-62"><a href="#cb105-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb105-63"><a href="#cb105-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb105-64"><a href="#cb105-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb105-65"><a href="#cb105-65" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb105-66"><a href="#cb105-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-67"><a href="#cb105-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb105-68"><a href="#cb105-68" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(small_intestine_cluster4_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = small_intestine_cluster4_SC, "small_intestine_cluster4_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="small-intestine-cluster-5---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="small-intestine-cluster-5---plot-heatmap">Small intestine cluster 5 - Plot heatmap</h2>
<section id="setup-for-analysis-4" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-4">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_small</span></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">5</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a> <span class="co"># dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">5</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb107-13"><a href="#cb107-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb107-14"><a href="#cb107-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb107-15"><a href="#cb107-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb107-16"><a href="#cb107-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-17"><a href="#cb107-17" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb107-18"><a href="#cb107-18" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb107-19"><a href="#cb107-19" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb107-20"><a href="#cb107-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-21"><a href="#cb107-21" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb107-22"><a href="#cb107-22" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb107-23"><a href="#cb107-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-24"><a href="#cb107-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb107-25"><a href="#cb107-25" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb107-26"><a href="#cb107-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-27"><a href="#cb107-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb107-28"><a href="#cb107-28" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_small_intestine_cluster5.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-5" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-5">Plot heatmap of cluster 5</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_small_intestine_cluster5.Rdata"</span>)</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb108-9"><a href="#cb108-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb108-10"><a href="#cb108-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb108-11"><a href="#cb108-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-12"><a href="#cb108-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb108-13"><a href="#cb108-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb108-14"><a href="#cb108-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb108-15"><a href="#cb108-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-16"><a href="#cb108-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb108-17"><a href="#cb108-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb108-18"><a href="#cb108-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb108-19"><a href="#cb108-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb108-20"><a href="#cb108-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb108-21"><a href="#cb108-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb108-22"><a href="#cb108-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb108-23"><a href="#cb108-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb108-24"><a href="#cb108-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-25"><a href="#cb108-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb108-26"><a href="#cb108-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-27"><a href="#cb108-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb108-28"><a href="#cb108-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb108-29"><a href="#cb108-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-30"><a href="#cb108-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb108-31"><a href="#cb108-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb108-32"><a href="#cb108-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-33"><a href="#cb108-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb108-34"><a href="#cb108-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-35"><a href="#cb108-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-36"><a href="#cb108-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-37"><a href="#cb108-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb108-38"><a href="#cb108-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb108-39"><a href="#cb108-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-40"><a href="#cb108-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-41"><a href="#cb108-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-42"><a href="#cb108-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb108-43"><a href="#cb108-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb108-44"><a href="#cb108-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-45"><a href="#cb108-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb108-46"><a href="#cb108-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb108-47"><a href="#cb108-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-48"><a href="#cb108-48" aria-hidden="true" tabindex="-1"></a>small_intestine_cluster5_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb108-49"><a href="#cb108-49" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb108-50"><a href="#cb108-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb108-51"><a href="#cb108-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb108-52"><a href="#cb108-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb108-53"><a href="#cb108-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb108-54"><a href="#cb108-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb108-55"><a href="#cb108-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb108-56"><a href="#cb108-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb108-57"><a href="#cb108-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb108-58"><a href="#cb108-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb108-59"><a href="#cb108-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb108-60"><a href="#cb108-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb108-61"><a href="#cb108-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb108-62"><a href="#cb108-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb108-63"><a href="#cb108-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb108-64"><a href="#cb108-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb108-65"><a href="#cb108-65" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb108-66"><a href="#cb108-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-67"><a href="#cb108-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb108-68"><a href="#cb108-68" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(small_intestine_cluster5_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = small_intestine_cluster5_SC, "small_intestine_cluster5_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="small-intestine-cluster-6---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="small-intestine-cluster-6---plot-heatmap">Small intestine cluster 6 - Plot heatmap</h2>
<section id="setup-for-analysis-5" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-5">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_small</span></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">6</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">6</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb110-15"><a href="#cb110-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb110-16"><a href="#cb110-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-17"><a href="#cb110-17" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb110-18"><a href="#cb110-18" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb110-19"><a href="#cb110-19" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb110-20"><a href="#cb110-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-21"><a href="#cb110-21" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb110-22"><a href="#cb110-22" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb110-23"><a href="#cb110-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-24"><a href="#cb110-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb110-25"><a href="#cb110-25" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb110-26"><a href="#cb110-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-27"><a href="#cb110-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb110-28"><a href="#cb110-28" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_small_intestine_cluster6.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-6" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-6">Plot heatmap of cluster 6</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_small_intestine_cluster6.Rdata"</span>)</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb111-18"><a href="#cb111-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb111-19"><a href="#cb111-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb111-20"><a href="#cb111-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb111-21"><a href="#cb111-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb111-22"><a href="#cb111-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb111-23"><a href="#cb111-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb111-24"><a href="#cb111-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-25"><a href="#cb111-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb111-26"><a href="#cb111-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-27"><a href="#cb111-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb111-28"><a href="#cb111-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb111-29"><a href="#cb111-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-30"><a href="#cb111-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb111-31"><a href="#cb111-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb111-32"><a href="#cb111-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-33"><a href="#cb111-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb111-34"><a href="#cb111-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-35"><a href="#cb111-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-36"><a href="#cb111-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-37"><a href="#cb111-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb111-38"><a href="#cb111-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb111-39"><a href="#cb111-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-40"><a href="#cb111-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-41"><a href="#cb111-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-42"><a href="#cb111-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb111-43"><a href="#cb111-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb111-44"><a href="#cb111-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-45"><a href="#cb111-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb111-46"><a href="#cb111-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb111-47"><a href="#cb111-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-48"><a href="#cb111-48" aria-hidden="true" tabindex="-1"></a>small_intestine_cluster6_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb111-49"><a href="#cb111-49" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb111-50"><a href="#cb111-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb111-51"><a href="#cb111-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb111-52"><a href="#cb111-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb111-53"><a href="#cb111-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb111-54"><a href="#cb111-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb111-55"><a href="#cb111-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb111-56"><a href="#cb111-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb111-57"><a href="#cb111-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb111-58"><a href="#cb111-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb111-59"><a href="#cb111-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb111-60"><a href="#cb111-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb111-61"><a href="#cb111-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb111-62"><a href="#cb111-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb111-63"><a href="#cb111-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb111-64"><a href="#cb111-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb111-65"><a href="#cb111-65" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb111-66"><a href="#cb111-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-67"><a href="#cb111-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb111-68"><a href="#cb111-68" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(small_intestine_cluster6_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = small_intestine_cluster6_SC, "small_intestine_cluster6_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="small-intestine-cluster-8---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="small-intestine-cluster-8---plot-heatmap">Small intestine cluster 8 - Plot heatmap</h2>
<section id="setup-for-analysis-6" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-6">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_small</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">8</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">8</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-23"><a href="#cb113-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb113-24"><a href="#cb113-24" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb113-25"><a href="#cb113-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-26"><a href="#cb113-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-27"><a href="#cb113-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb113-28"><a href="#cb113-28" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_small_intestine_cluster8.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-8" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-8">Plot heatmap of cluster 8</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_small_intestine_cluster8.Rdata"</span>)</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb114-23"><a href="#cb114-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb114-24"><a href="#cb114-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-25"><a href="#cb114-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb114-26"><a href="#cb114-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-27"><a href="#cb114-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb114-28"><a href="#cb114-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb114-29"><a href="#cb114-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-30"><a href="#cb114-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb114-31"><a href="#cb114-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb114-32"><a href="#cb114-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-33"><a href="#cb114-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb114-34"><a href="#cb114-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-35"><a href="#cb114-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-36"><a href="#cb114-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-37"><a href="#cb114-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb114-38"><a href="#cb114-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb114-39"><a href="#cb114-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-40"><a href="#cb114-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-41"><a href="#cb114-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-42"><a href="#cb114-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb114-43"><a href="#cb114-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb114-44"><a href="#cb114-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-45"><a href="#cb114-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb114-46"><a href="#cb114-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb114-47"><a href="#cb114-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-48"><a href="#cb114-48" aria-hidden="true" tabindex="-1"></a>small_intestine_cluster8_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb114-49"><a href="#cb114-49" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb114-50"><a href="#cb114-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb114-51"><a href="#cb114-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb114-52"><a href="#cb114-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb114-53"><a href="#cb114-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb114-54"><a href="#cb114-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb114-55"><a href="#cb114-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb114-56"><a href="#cb114-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb114-57"><a href="#cb114-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb114-58"><a href="#cb114-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb114-59"><a href="#cb114-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb114-60"><a href="#cb114-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb114-61"><a href="#cb114-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb114-62"><a href="#cb114-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb114-63"><a href="#cb114-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb114-64"><a href="#cb114-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb114-65"><a href="#cb114-65" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb114-66"><a href="#cb114-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-67"><a href="#cb114-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb114-68"><a href="#cb114-68" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(small_intestine_cluster8_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = small_intestine_cluster8_SC, "small_intestine_cluster8_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>#Plot heatmap of cluster in the healthy large intestine ## Large intestine cluster 1 - Plot heatmap</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Library/Mobile Documents/com~apple~CloudDocs/pca_project/R/heat.cluster.large.Rdata"</span>)</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Library/Mobile Documents/com~apple~CloudDocs/pca_project/R/heat.clust.go_large_control.Rdata"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="setup-for-analysis-7" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-7">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_Large</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = 100) %&gt;%</span></span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_large[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb117-17"><a href="#cb117-17" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb117-18"><a href="#cb117-18" aria-hidden="true" tabindex="-1"></a><span class="co">#heat$variable=gsub("[.]","-",heat$variable)</span></span>
<span id="cb117-19"><a href="#cb117-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-20"><a href="#cb117-20" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb117-21"><a href="#cb117-21" aria-hidden="true" tabindex="-1"></a>umap_info_large<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_large)</span>
<span id="cb117-22"><a href="#cb117-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-23"><a href="#cb117-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb117-24"><a href="#cb117-24" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_large[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb117-25"><a href="#cb117-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-26"><a href="#cb117-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb117-27"><a href="#cb117-27" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_large_intestine_cluster1.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-1" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-1">Plot heatmap of cluster 1</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_large_intestine_cluster1.Rdata"</span>)</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-12"><a href="#cb118-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb118-13"><a href="#cb118-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb118-14"><a href="#cb118-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb118-15"><a href="#cb118-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-16"><a href="#cb118-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb118-17"><a href="#cb118-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb118-18"><a href="#cb118-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb118-19"><a href="#cb118-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb118-20"><a href="#cb118-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb118-21"><a href="#cb118-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb118-22"><a href="#cb118-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb118-23"><a href="#cb118-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb118-24"><a href="#cb118-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-25"><a href="#cb118-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb118-26"><a href="#cb118-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-27"><a href="#cb118-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb118-28"><a href="#cb118-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb118-29"><a href="#cb118-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-30"><a href="#cb118-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb118-31"><a href="#cb118-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_large]</span>
<span id="cb118-32"><a href="#cb118-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-33"><a href="#cb118-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb118-34"><a href="#cb118-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-35"><a href="#cb118-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-36"><a href="#cb118-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-37"><a href="#cb118-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb118-38"><a href="#cb118-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb118-39"><a href="#cb118-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-40"><a href="#cb118-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-41"><a href="#cb118-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-42"><a href="#cb118-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb118-43"><a href="#cb118-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb118-44"><a href="#cb118-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-45"><a href="#cb118-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb118-46"><a href="#cb118-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb118-47"><a href="#cb118-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-48"><a href="#cb118-48" aria-hidden="true" tabindex="-1"></a>large_intestine_cluster1_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb118-49"><a href="#cb118-49" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb118-50"><a href="#cb118-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb118-51"><a href="#cb118-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb118-52"><a href="#cb118-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb118-53"><a href="#cb118-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb118-54"><a href="#cb118-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_large,</span>
<span id="cb118-55"><a href="#cb118-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb118-56"><a href="#cb118-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb118-57"><a href="#cb118-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb118-58"><a href="#cb118-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb118-59"><a href="#cb118-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb118-60"><a href="#cb118-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb118-61"><a href="#cb118-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb118-62"><a href="#cb118-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb118-63"><a href="#cb118-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb118-64"><a href="#cb118-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb118-65"><a href="#cb118-65" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb118-66"><a href="#cb118-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-67"><a href="#cb118-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb118-68"><a href="#cb118-68" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(large_intestine_cluster1_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = large_intestine_cluster1_SC, "large_intestine_cluster1_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="large-intestine-cluster-2---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="large-intestine-cluster-2---plot-heatmap">Large intestine cluster 2 - Plot heatmap</h2>
<section id="setup-for-analysis-8" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-8">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_Large</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = 100) %&gt;%</span></span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-9"><a href="#cb120-9" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb120-10"><a href="#cb120-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb120-11"><a href="#cb120-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb120-12"><a href="#cb120-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb120-13"><a href="#cb120-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb120-14"><a href="#cb120-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-15"><a href="#cb120-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-16"><a href="#cb120-16" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_large[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb120-17"><a href="#cb120-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb120-18"><a href="#cb120-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb120-19"><a href="#cb120-19" aria-hidden="true" tabindex="-1"></a><span class="co">#heat$variable=gsub("[.]","-",heat$variable)</span></span>
<span id="cb120-20"><a href="#cb120-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-21"><a href="#cb120-21" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb120-22"><a href="#cb120-22" aria-hidden="true" tabindex="-1"></a>umap_info_large<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_large)</span>
<span id="cb120-23"><a href="#cb120-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-24"><a href="#cb120-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb120-25"><a href="#cb120-25" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_large[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb120-26"><a href="#cb120-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-27"><a href="#cb120-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb120-28"><a href="#cb120-28" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(heat, <span class="at">file =</span> <span class="st">"heatmap_large_intestine_cluster2.Rdata"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-2-1" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-2-1">Plot heatmap of cluster 2</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_large_intestine_cluster2.Rdata"</span>)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb121-14"><a href="#cb121-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb121-15"><a href="#cb121-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-16"><a href="#cb121-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb121-17"><a href="#cb121-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb121-18"><a href="#cb121-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb121-19"><a href="#cb121-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb121-20"><a href="#cb121-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb121-21"><a href="#cb121-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb121-22"><a href="#cb121-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb121-23"><a href="#cb121-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb121-24"><a href="#cb121-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-25"><a href="#cb121-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb121-26"><a href="#cb121-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-27"><a href="#cb121-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb121-28"><a href="#cb121-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb121-29"><a href="#cb121-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-30"><a href="#cb121-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb121-31"><a href="#cb121-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_large]</span>
<span id="cb121-32"><a href="#cb121-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-33"><a href="#cb121-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb121-34"><a href="#cb121-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-35"><a href="#cb121-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-36"><a href="#cb121-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-37"><a href="#cb121-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb121-38"><a href="#cb121-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb121-39"><a href="#cb121-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-40"><a href="#cb121-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-41"><a href="#cb121-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-42"><a href="#cb121-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb121-43"><a href="#cb121-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb121-44"><a href="#cb121-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-45"><a href="#cb121-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb121-46"><a href="#cb121-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb121-47"><a href="#cb121-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-48"><a href="#cb121-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-49"><a href="#cb121-49" aria-hidden="true" tabindex="-1"></a>large_intestine_cluster2_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb121-50"><a href="#cb121-50" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb121-51"><a href="#cb121-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb121-52"><a href="#cb121-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb121-53"><a href="#cb121-53" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb121-54"><a href="#cb121-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb121-55"><a href="#cb121-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_large,</span>
<span id="cb121-56"><a href="#cb121-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb121-57"><a href="#cb121-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb121-58"><a href="#cb121-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb121-59"><a href="#cb121-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb121-60"><a href="#cb121-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb121-61"><a href="#cb121-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb121-62"><a href="#cb121-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb121-63"><a href="#cb121-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb121-64"><a href="#cb121-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb121-65"><a href="#cb121-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb121-66"><a href="#cb121-66" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb121-67"><a href="#cb121-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-68"><a href="#cb121-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb121-69"><a href="#cb121-69" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(large_intestine_cluster2_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = large_intestine_cluster2_SC, "large_intestine_cluster2_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="large-intestine-cluster-5---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="large-intestine-cluster-5---plot-heatmap">Large intestine cluster 5 - Plot heatmap</h2>
<section id="setup-for-analysis-9" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-9">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_Large</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">5</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = 100) %&gt;%</span></span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-9"><a href="#cb123-9" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">5</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb123-10"><a href="#cb123-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb123-11"><a href="#cb123-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb123-12"><a href="#cb123-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb123-13"><a href="#cb123-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb123-14"><a href="#cb123-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-15"><a href="#cb123-15" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_large[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb123-16"><a href="#cb123-16" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb123-17"><a href="#cb123-17" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb123-18"><a href="#cb123-18" aria-hidden="true" tabindex="-1"></a><span class="co">#heat$variable=gsub("[.]","-",heat$variable)</span></span>
<span id="cb123-19"><a href="#cb123-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-20"><a href="#cb123-20" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb123-21"><a href="#cb123-21" aria-hidden="true" tabindex="-1"></a>umap_info_large<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_large)</span>
<span id="cb123-22"><a href="#cb123-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-23"><a href="#cb123-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb123-24"><a href="#cb123-24" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_large[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb123-25"><a href="#cb123-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-26"><a href="#cb123-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb123-27"><a href="#cb123-27" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(heat, <span class="at">file =</span> <span class="st">"heatmap_large_intestine_cluster5.Rdata"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-5-1" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-5-1">Plot heatmap of cluster 5</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_large_intestine_cluster5.Rdata"</span>)</span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb124-13"><a href="#cb124-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb124-14"><a href="#cb124-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb124-15"><a href="#cb124-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-16"><a href="#cb124-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb124-17"><a href="#cb124-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb124-18"><a href="#cb124-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb124-19"><a href="#cb124-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb124-20"><a href="#cb124-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb124-21"><a href="#cb124-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb124-22"><a href="#cb124-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb124-23"><a href="#cb124-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb124-24"><a href="#cb124-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-25"><a href="#cb124-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb124-26"><a href="#cb124-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-27"><a href="#cb124-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb124-28"><a href="#cb124-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb124-29"><a href="#cb124-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-30"><a href="#cb124-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb124-31"><a href="#cb124-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_large]</span>
<span id="cb124-32"><a href="#cb124-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-33"><a href="#cb124-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb124-34"><a href="#cb124-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-35"><a href="#cb124-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-36"><a href="#cb124-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-37"><a href="#cb124-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb124-38"><a href="#cb124-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb124-39"><a href="#cb124-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-40"><a href="#cb124-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-41"><a href="#cb124-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-42"><a href="#cb124-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb124-43"><a href="#cb124-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb124-44"><a href="#cb124-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-45"><a href="#cb124-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb124-46"><a href="#cb124-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb124-47"><a href="#cb124-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-48"><a href="#cb124-48" aria-hidden="true" tabindex="-1"></a>large_intestine_cluster5_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb124-49"><a href="#cb124-49" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb124-50"><a href="#cb124-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb124-51"><a href="#cb124-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb124-52"><a href="#cb124-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb124-53"><a href="#cb124-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb124-54"><a href="#cb124-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_large,</span>
<span id="cb124-55"><a href="#cb124-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb124-56"><a href="#cb124-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb124-57"><a href="#cb124-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb124-58"><a href="#cb124-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb124-59"><a href="#cb124-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb124-60"><a href="#cb124-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb124-61"><a href="#cb124-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb124-62"><a href="#cb124-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb124-63"><a href="#cb124-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb124-64"><a href="#cb124-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb124-65"><a href="#cb124-65" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb124-66"><a href="#cb124-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-67"><a href="#cb124-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb124-68"><a href="#cb124-68" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(large_intestine_cluster5_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = large_intestine_cluster5_SC, "large_intestine_cluster5_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="large-intestine-cluster-8---plot-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="large-intestine-cluster-8---plot-heatmap">Large intestine cluster 8 - Plot heatmap</h2>
<section id="setup-for-analysis-10" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-10">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_Large</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> heat.cluster.<span class="sc">$</span><span class="st">`</span><span class="at">8</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">enframe</span>(<span class="at">name =</span> <span class="cn">NULL</span>, <span class="at">value =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = 100) %&gt;%</span></span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL =</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">gsub</span>(<span class="st">"/"</span>, <span class="st">" "</span>, heat.clust.go.<span class="sc">$</span><span class="st">`</span><span class="at">8</span><span class="st">`</span>[<span class="dv">1</span>, <span class="st">"geneID"</span>]), <span class="st">"</span><span class="sc">\\</span><span class="st">s"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb126-10"><a href="#cb126-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene2acc, <span class="at">by =</span> <span class="st">"ENSEMBL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb126-11"><a href="#cb126-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(hgnc_symbol, <span class="at">.after =</span> ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb126-12"><a href="#cb126-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::slice_head(n = number_of_genes) %&gt;%</span></span>
<span id="cb126-13"><a href="#cb126-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb126-14"><a href="#cb126-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-15"><a href="#cb126-15" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_large[genes_to_plot,]<span class="co">#Subsetting expression dataframe of genes of interest (goi)</span></span>
<span id="cb126-16"><a href="#cb126-16" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb126-17"><a href="#cb126-17" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb126-18"><a href="#cb126-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-19"><a href="#cb126-19" aria-hidden="true" tabindex="-1"></a><span class="co"># First, add the cell ID as a column if it's in the rownames</span></span>
<span id="cb126-20"><a href="#cb126-20" aria-hidden="true" tabindex="-1"></a>umap_info_large<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_large)</span>
<span id="cb126-21"><a href="#cb126-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-22"><a href="#cb126-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Now merge should work</span></span>
<span id="cb126-23"><a href="#cb126-23" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_large[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb126-24"><a href="#cb126-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-25"><a href="#cb126-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb126-26"><a href="#cb126-26" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_large_intestine_cluster8.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-cluster-8-1" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-cluster-8-1">Plot heatmap of cluster 8</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_large_intestine_cluster8.Rdata"</span>)</span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb127-5"><a href="#cb127-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb127-6"><a href="#cb127-6" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb127-7"><a href="#cb127-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb127-8"><a href="#cb127-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb127-9"><a href="#cb127-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb127-10"><a href="#cb127-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb127-11"><a href="#cb127-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-12"><a href="#cb127-12" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb127-13"><a href="#cb127-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb127-14"><a href="#cb127-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb127-15"><a href="#cb127-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-16"><a href="#cb127-16" aria-hidden="true" tabindex="-1"></a><span class="co">#We can´t scale genes if they are not detected. So remove all genes which have an average of 0 in all cells</span></span>
<span id="cb127-17"><a href="#cb127-17" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb127-18"><a href="#cb127-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb127-19"><a href="#cb127-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb127-20"><a href="#cb127-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance (to avoid NaNs when scaling)</span></span>
<span id="cb127-21"><a href="#cb127-21" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb127-22"><a href="#cb127-22" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb127-23"><a href="#cb127-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb127-24"><a href="#cb127-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-25"><a href="#cb127-25" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb127-26"><a href="#cb127-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-27"><a href="#cb127-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb127-28"><a href="#cb127-28" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb127-29"><a href="#cb127-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-30"><a href="#cb127-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-31"><a href="#cb127-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb127-32"><a href="#cb127-32" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_large]</span>
<span id="cb127-33"><a href="#cb127-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-34"><a href="#cb127-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb127-35"><a href="#cb127-35" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-36"><a href="#cb127-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-37"><a href="#cb127-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-38"><a href="#cb127-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb127-39"><a href="#cb127-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb127-40"><a href="#cb127-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-41"><a href="#cb127-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-42"><a href="#cb127-42" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-43"><a href="#cb127-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb127-44"><a href="#cb127-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb127-45"><a href="#cb127-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-46"><a href="#cb127-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb127-47"><a href="#cb127-47" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb127-48"><a href="#cb127-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-49"><a href="#cb127-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-50"><a href="#cb127-50" aria-hidden="true" tabindex="-1"></a>large_intestine_cluster8_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb127-51"><a href="#cb127-51" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb127-52"><a href="#cb127-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb127-53"><a href="#cb127-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb127-54"><a href="#cb127-54" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb127-55"><a href="#cb127-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb127-56"><a href="#cb127-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_large,</span>
<span id="cb127-57"><a href="#cb127-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb127-58"><a href="#cb127-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb127-59"><a href="#cb127-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb127-60"><a href="#cb127-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb127-61"><a href="#cb127-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb127-62"><a href="#cb127-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb127-63"><a href="#cb127-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb127-64"><a href="#cb127-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb127-65"><a href="#cb127-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb127-66"><a href="#cb127-66" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span> <span class="co"># Rotate column labels</span></span>
<span id="cb127-67"><a href="#cb127-67" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb127-68"><a href="#cb127-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-69"><a href="#cb127-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb127-70"><a href="#cb127-70" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(large_intestine_cluster8_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = large_intestine_cluster8_SC, "large_intestine_cluster8_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="plot-degs-in-single-cells" class="level1">
<h1>Plot DEGs in single cells</h1>
<section id="small-intestines---heatmap" class="level2">
<h2 class="anchored" data-anchor-id="small-intestines---heatmap">Small intestines - Heatmap</h2>
<section id="setup-for-analysis-11" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-11">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_small</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> res_df_small <span class="sc">%&gt;%</span></span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> number_of_genes) <span class="sc">%&gt;%</span></span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">hgnc_symbol =</span> SYMBOL) <span class="sc">%&gt;%</span></span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hgnc_symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(expression_small)) <span class="sc">%&gt;%</span></span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(hgnc_symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb129-11"><a href="#cb129-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-12"><a href="#cb129-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Subsetting expression dataframe of genes of interest</span></span>
<span id="cb129-13"><a href="#cb129-13" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]</span>
<span id="cb129-14"><a href="#cb129-14" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb129-15"><a href="#cb129-15" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>) <span class="co">#Reformatting</span></span>
<span id="cb129-16"><a href="#cb129-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-17"><a href="#cb129-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the cell ID as a column if it's in the rownames</span></span>
<span id="cb129-18"><a href="#cb129-18" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb129-19"><a href="#cb129-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-20"><a href="#cb129-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge</span></span>
<span id="cb129-21"><a href="#cb129-21" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb129-22"><a href="#cb129-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-23"><a href="#cb129-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb129-24"><a href="#cb129-24" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_small_intestine_DEG.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-degs-in-small-intestines" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-degs-in-small-intestines">Plot heatmap of DEGs in small intestines</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_small_intestine_DEG.Rdata"</span>)</span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by ident and gene, and calculate the mean value for each gene in each cell type</span></span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-15"><a href="#cb130-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the results into a single data frame</span></span>
<span id="cb130-16"><a href="#cb130-16" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb130-17"><a href="#cb130-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb130-18"><a href="#cb130-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb130-19"><a href="#cb130-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-20"><a href="#cb130-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Remove all genes which have an average of 0 in all cells</span></span>
<span id="cb130-21"><a href="#cb130-21" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb130-22"><a href="#cb130-22" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb130-23"><a href="#cb130-23" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb130-24"><a href="#cb130-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-25"><a href="#cb130-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance</span></span>
<span id="cb130-26"><a href="#cb130-26" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb130-27"><a href="#cb130-27" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb130-28"><a href="#cb130-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-29"><a href="#cb130-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb130-30"><a href="#cb130-30" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb130-31"><a href="#cb130-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-32"><a href="#cb130-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb130-33"><a href="#cb130-33" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb130-34"><a href="#cb130-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-35"><a href="#cb130-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder rows based on the mean expression across all cell types</span></span>
<span id="cb130-36"><a href="#cb130-36" aria-hidden="true" tabindex="-1"></a>gene_colors <span class="ot">&lt;-</span> res_df_small <span class="sc">%&gt;%</span></span>
<span id="cb130-37"><a href="#cb130-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb130-38"><a href="#cb130-38" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb130-39"><a href="#cb130-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> number_of_genes) <span class="sc">%&gt;%</span></span>
<span id="cb130-40"><a href="#cb130-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">hgnc_symbol =</span> SYMBOL) <span class="sc">%&gt;%</span></span>
<span id="cb130-41"><a href="#cb130-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(hgnc_symbol, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb130-42"><a href="#cb130-42" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">ifelse</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Upregulated"</span>, <span class="st">"Downregulated"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb130-43"><a href="#cb130-43" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hgnc_symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(heat)) <span class="sc">%&gt;%</span></span>
<span id="cb130-44"><a href="#cb130-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(hgnc_symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb130-45"><a href="#cb130-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-46"><a href="#cb130-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation row with labels</span></span>
<span id="cb130-47"><a href="#cb130-47" aria-hidden="true" tabindex="-1"></a>annotation_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Regulation =</span> gene_colors<span class="sc">$</span>direction)</span>
<span id="cb130-48"><a href="#cb130-48" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation_row) <span class="ot">&lt;-</span> gene_colors<span class="sc">$</span>hgnc_symbol</span>
<span id="cb130-49"><a href="#cb130-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-50"><a href="#cb130-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation row with labels</span></span>
<span id="cb130-51"><a href="#cb130-51" aria-hidden="true" tabindex="-1"></a>annotation_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Direction =</span> gene_colors<span class="sc">$</span>direction)</span>
<span id="cb130-52"><a href="#cb130-52" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation_row) <span class="ot">&lt;-</span> gene_colors<span class="sc">$</span>hgnc_symbol</span>
<span id="cb130-53"><a href="#cb130-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-54"><a href="#cb130-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a named vector for annotation colors</span></span>
<span id="cb130-55"><a href="#cb130-55" aria-hidden="true" tabindex="-1"></a>annotation_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Direction =</span> <span class="fu">c</span>(<span class="st">"Upregulated"</span> <span class="ot">=</span> <span class="st">"#b2182b"</span>, <span class="st">"Downregulated"</span> <span class="ot">=</span> <span class="st">"#2166ac"</span>))</span>
<span id="cb130-56"><a href="#cb130-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-57"><a href="#cb130-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb130-58"><a href="#cb130-58" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb130-59"><a href="#cb130-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-60"><a href="#cb130-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb130-61"><a href="#cb130-61" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-62"><a href="#cb130-62" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-63"><a href="#cb130-63" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-64"><a href="#cb130-64" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb130-65"><a href="#cb130-65" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb130-66"><a href="#cb130-66" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-67"><a href="#cb130-67" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-68"><a href="#cb130-68" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-69"><a href="#cb130-69" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb130-70"><a href="#cb130-70" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb130-71"><a href="#cb130-71" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-72"><a href="#cb130-72" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb130-73"><a href="#cb130-73" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb130-74"><a href="#cb130-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-75"><a href="#cb130-75" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb130-76"><a href="#cb130-76" aria-hidden="true" tabindex="-1"></a>small_intestines_DEG_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb130-77"><a href="#cb130-77" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb130-78"><a href="#cb130-78" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb130-79"><a href="#cb130-79" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb130-80"><a href="#cb130-80" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb130-81"><a href="#cb130-81" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb130-82"><a href="#cb130-82" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb130-83"><a href="#cb130-83" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb130-84"><a href="#cb130-84" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb130-85"><a href="#cb130-85" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb130-86"><a href="#cb130-86" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb130-87"><a href="#cb130-87" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb130-88"><a href="#cb130-88" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb130-89"><a href="#cb130-89" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb130-90"><a href="#cb130-90" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb130-91"><a href="#cb130-91" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb130-92"><a href="#cb130-92" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span>, <span class="co"># Rotate column labels</span></span>
<span id="cb130-93"><a href="#cb130-93" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_row =</span> annotation_row,</span>
<span id="cb130-94"><a href="#cb130-94" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> annotation_colors</span>
<span id="cb130-95"><a href="#cb130-95" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb130-96"><a href="#cb130-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-97"><a href="#cb130-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb130-98"><a href="#cb130-98" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(small_intestines_DEG_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = small_intestines_DEG_SC, "small_intestines_DEG_SC",width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="distal-small-intestines---heatmap" class="level2">
<h2 class="anchored" data-anchor-id="distal-small-intestines---heatmap">Distal small intestines - Heatmap</h2>
<section id="setup-for-analysis-12" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-12">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_ileocecal</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> res_df_ileocecal <span class="sc">%&gt;%</span></span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> number_of_genes) <span class="sc">%&gt;%</span></span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">hgnc_symbol =</span> SYMBOL) <span class="sc">%&gt;%</span></span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hgnc_symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(expression_small)) <span class="sc">%&gt;%</span></span>
<span id="cb132-9"><a href="#cb132-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(hgnc_symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb132-10"><a href="#cb132-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb132-11"><a href="#cb132-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-12"><a href="#cb132-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Subsetting expression dataframe of genes of interest</span></span>
<span id="cb132-13"><a href="#cb132-13" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_small[genes_to_plot,]</span>
<span id="cb132-14"><a href="#cb132-14" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb132-15"><a href="#cb132-15" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>) <span class="co">#Reformatting</span></span>
<span id="cb132-16"><a href="#cb132-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-17"><a href="#cb132-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the cell ID as a column if it's in the rownames</span></span>
<span id="cb132-18"><a href="#cb132-18" aria-hidden="true" tabindex="-1"></a>umap_info_small<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_small)</span>
<span id="cb132-19"><a href="#cb132-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-20"><a href="#cb132-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge</span></span>
<span id="cb132-21"><a href="#cb132-21" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_small[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb132-22"><a href="#cb132-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-23"><a href="#cb132-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb132-24"><a href="#cb132-24" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_distal_small_intestine_DEG.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-degs-in-distal-small-intestines" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-degs-in-distal-small-intestines">Plot heatmap of DEGs in distal small intestines</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">75</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_distal_small_intestine_DEG.Rdata"</span>)</span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by ident and gene, and calculate the mean value for each gene in each cell type</span></span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb133-10"><a href="#cb133-10" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb133-11"><a href="#cb133-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb133-12"><a href="#cb133-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb133-13"><a href="#cb133-13" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb133-14"><a href="#cb133-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-15"><a href="#cb133-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the results into a single data frame</span></span>
<span id="cb133-16"><a href="#cb133-16" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb133-17"><a href="#cb133-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb133-18"><a href="#cb133-18" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb133-19"><a href="#cb133-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-20"><a href="#cb133-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Remove all genes which have an average of 0 in all cells</span></span>
<span id="cb133-21"><a href="#cb133-21" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb133-22"><a href="#cb133-22" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb133-23"><a href="#cb133-23" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb133-24"><a href="#cb133-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-25"><a href="#cb133-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance</span></span>
<span id="cb133-26"><a href="#cb133-26" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb133-27"><a href="#cb133-27" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb133-28"><a href="#cb133-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-29"><a href="#cb133-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb133-30"><a href="#cb133-30" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb133-31"><a href="#cb133-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-32"><a href="#cb133-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb133-33"><a href="#cb133-33" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb133-34"><a href="#cb133-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-35"><a href="#cb133-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder rows based on the mean expression across all cell types</span></span>
<span id="cb133-36"><a href="#cb133-36" aria-hidden="true" tabindex="-1"></a>gene_colors <span class="ot">&lt;-</span> res_df_ileocecal <span class="sc">%&gt;%</span></span>
<span id="cb133-37"><a href="#cb133-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb133-38"><a href="#cb133-38" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb133-39"><a href="#cb133-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> number_of_genes) <span class="sc">%&gt;%</span></span>
<span id="cb133-40"><a href="#cb133-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">hgnc_symbol =</span> SYMBOL) <span class="sc">%&gt;%</span></span>
<span id="cb133-41"><a href="#cb133-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(hgnc_symbol, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb133-42"><a href="#cb133-42" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">ifelse</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Upregulated"</span>, <span class="st">"Downregulated"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb133-43"><a href="#cb133-43" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hgnc_symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(heat)) <span class="sc">%&gt;%</span></span>
<span id="cb133-44"><a href="#cb133-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(hgnc_symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb133-45"><a href="#cb133-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-46"><a href="#cb133-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation row with labels</span></span>
<span id="cb133-47"><a href="#cb133-47" aria-hidden="true" tabindex="-1"></a>annotation_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Direction =</span> gene_colors<span class="sc">$</span>direction)</span>
<span id="cb133-48"><a href="#cb133-48" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation_row) <span class="ot">&lt;-</span> gene_colors<span class="sc">$</span>hgnc_symbol</span>
<span id="cb133-49"><a href="#cb133-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-50"><a href="#cb133-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a named vector for annotation colors</span></span>
<span id="cb133-51"><a href="#cb133-51" aria-hidden="true" tabindex="-1"></a>annotation_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Direction =</span> <span class="fu">c</span>(<span class="st">"Upregulated"</span> <span class="ot">=</span> <span class="st">"#b2182b"</span>, <span class="st">"Downregulated"</span> <span class="ot">=</span> <span class="st">"#2166ac"</span>))</span>
<span id="cb133-52"><a href="#cb133-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-53"><a href="#cb133-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb133-54"><a href="#cb133-54" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_small]</span>
<span id="cb133-55"><a href="#cb133-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-56"><a href="#cb133-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb133-57"><a href="#cb133-57" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-58"><a href="#cb133-58" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-59"><a href="#cb133-59" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-60"><a href="#cb133-60" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb133-61"><a href="#cb133-61" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb133-62"><a href="#cb133-62" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-63"><a href="#cb133-63" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-64"><a href="#cb133-64" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-65"><a href="#cb133-65" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb133-66"><a href="#cb133-66" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb133-67"><a href="#cb133-67" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-68"><a href="#cb133-68" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb133-69"><a href="#cb133-69" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb133-70"><a href="#cb133-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-71"><a href="#cb133-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb133-72"><a href="#cb133-72" aria-hidden="true" tabindex="-1"></a>ileocecal_DEG_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb133-73"><a href="#cb133-73" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb133-74"><a href="#cb133-74" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb133-75"><a href="#cb133-75" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb133-76"><a href="#cb133-76" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb133-77"><a href="#cb133-77" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb133-78"><a href="#cb133-78" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_small,</span>
<span id="cb133-79"><a href="#cb133-79" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb133-80"><a href="#cb133-80" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb133-81"><a href="#cb133-81" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb133-82"><a href="#cb133-82" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb133-83"><a href="#cb133-83" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb133-84"><a href="#cb133-84" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb133-85"><a href="#cb133-85" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb133-86"><a href="#cb133-86" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb133-87"><a href="#cb133-87" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb133-88"><a href="#cb133-88" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span>, <span class="co"># Rotate column labels</span></span>
<span id="cb133-89"><a href="#cb133-89" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_row =</span> annotation_row,</span>
<span id="cb133-90"><a href="#cb133-90" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> annotation_colors</span>
<span id="cb133-91"><a href="#cb133-91" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb133-92"><a href="#cb133-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-93"><a href="#cb133-93" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb133-94"><a href="#cb133-94" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(ileocecal_DEG_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a><span class="co"># graph2ppt(x = ileocecal_DEG_SC, "ileocecal_DEG_SC", width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="large-intestines---heatmap" class="level2">
<h2 class="anchored" data-anchor-id="large-intestines---heatmap">Large intestines - Heatmap</h2>
<section id="setup-for-analysis-13" class="level3">
<h3 class="anchored" data-anchor-id="setup-for-analysis-13">Setup for analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join HGNC symbols to res_df_large</span></span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">81</span> <span class="co"># Some are dublicates and will be removed later, the final figure will have 75 genes</span></span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> res_df_large <span class="sc">%&gt;%</span></span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> number_of_genes) <span class="sc">%&gt;%</span></span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">hgnc_symbol =</span> SYMBOL) <span class="sc">%&gt;%</span></span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hgnc_symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(expression_large)) <span class="sc">%&gt;%</span></span>
<span id="cb135-9"><a href="#cb135-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(hgnc_symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb135-10"><a href="#cb135-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(hgnc_symbol)</span>
<span id="cb135-11"><a href="#cb135-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-12"><a href="#cb135-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Subsetting expression dataframe of genes of interest</span></span>
<span id="cb135-13"><a href="#cb135-13" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>expression_large[genes_to_plot,]</span>
<span id="cb135-14"><a href="#cb135-14" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>gene<span class="ot">=</span><span class="fu">rownames</span>(heat)</span>
<span id="cb135-15"><a href="#cb135-15" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>reshape2<span class="sc">::</span><span class="fu">melt</span>(heat, <span class="at">id.vars=</span><span class="st">"gene"</span>)<span class="co">#Reformatting</span></span>
<span id="cb135-16"><a href="#cb135-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-17"><a href="#cb135-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the cell ID as a column if it's in the rownames</span></span>
<span id="cb135-18"><a href="#cb135-18" aria-hidden="true" tabindex="-1"></a>umap_info_large<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(umap_info_large)</span>
<span id="cb135-19"><a href="#cb135-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-20"><a href="#cb135-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge</span></span>
<span id="cb135-21"><a href="#cb135-21" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">=</span> <span class="fu">merge</span>(heat, umap_info_large[,<span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"cell"</span>)], <span class="at">by.x=</span><span class="st">"variable"</span>, <span class="at">by.y=</span><span class="st">"cell"</span>)</span>
<span id="cb135-22"><a href="#cb135-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-23"><a href="#cb135-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Save the heatmap data</span></span>
<span id="cb135-24"><a href="#cb135-24" aria-hidden="true" tabindex="-1"></a><span class="co">#save(heat, file = "heatmap_large_intestine_DEG.Rdata")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-heatmap-of-degs-in-large-intestines" class="level3">
<h3 class="anchored" data-anchor-id="plot-heatmap-of-degs-in-large-intestines">Plot heatmap of DEGs in large intestines</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>number_of_genes <span class="ot">&lt;-</span> <span class="dv">81</span> <span class="co"># Some are dublicates and will be removed later, the final figure will have 75 genes</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the heatmap data</span></span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"heatmap_large_intestine_DEG.Rdata"</span>)</span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by ident and gene, and calculate the mean value for each gene in each cell type</span></span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">lapply</span>(<span class="fu">unique</span>(heat<span class="sc">$</span>ident), <span class="cf">function</span>(x){</span>
<span id="cb136-7"><a href="#cb136-7" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>heat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ident<span class="sc">==</span>x)</span>
<span id="cb136-8"><a href="#cb136-8" aria-hidden="true" tabindex="-1"></a>  tmp<span class="ot">=</span>tmp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb136-9"><a href="#cb136-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tmp)[<span class="dv">2</span>]<span class="ot">=</span>x</span>
<span id="cb136-10"><a href="#cb136-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tmp)</span>
<span id="cb136-11"><a href="#cb136-11" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb136-12"><a href="#cb136-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-13"><a href="#cb136-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the results into a single data frame</span></span>
<span id="cb136-14"><a href="#cb136-14" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">do.call</span>(cbind,heat)</span>
<span id="cb136-15"><a href="#cb136-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat)<span class="ot">=</span>heat<span class="sc">$</span>gene</span>
<span id="cb136-16"><a href="#cb136-16" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[,<span class="fu">colnames</span>(heat) <span class="sc">!=</span> <span class="st">"gene"</span>]</span>
<span id="cb136-17"><a href="#cb136-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-18"><a href="#cb136-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Remove all genes which have an average of 0 in all cells</span></span>
<span id="cb136-19"><a href="#cb136-19" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="fu">apply</span>(heat, <span class="dv">1</span>,max) </span>
<span id="cb136-20"><a href="#cb136-20" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span>heat[heat<span class="sc">$</span>max <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb136-21"><a href="#cb136-21" aria-hidden="true" tabindex="-1"></a>heat<span class="sc">$</span>max<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb136-22"><a href="#cb136-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-23"><a href="#cb136-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with zero variance</span></span>
<span id="cb136-24"><a href="#cb136-24" aria-hidden="true" tabindex="-1"></a>row_sd <span class="ot">&lt;-</span> <span class="fu">apply</span>(heat, <span class="dv">1</span>, sd)</span>
<span id="cb136-25"><a href="#cb136-25" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[row_sd <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb136-26"><a href="#cb136-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-27"><a href="#cb136-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale</span></span>
<span id="cb136-28"><a href="#cb136-28" aria-hidden="true" tabindex="-1"></a>heat<span class="ot">=</span><span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heat)))</span>
<span id="cb136-29"><a href="#cb136-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-30"><a href="#cb136-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows that became NA after scaling</span></span>
<span id="cb136-31"><a href="#cb136-31" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[<span class="fu">complete.cases</span>(heat), ]</span>
<span id="cb136-32"><a href="#cb136-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-33"><a href="#cb136-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder rows based on the mean expression across all cell types</span></span>
<span id="cb136-34"><a href="#cb136-34" aria-hidden="true" tabindex="-1"></a>gene_colors <span class="ot">&lt;-</span> res_df_large <span class="sc">%&gt;%</span></span>
<span id="cb136-35"><a href="#cb136-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb136-36"><a href="#cb136-36" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb136-37"><a href="#cb136-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> number_of_genes) <span class="sc">%&gt;%</span></span>
<span id="cb136-38"><a href="#cb136-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">hgnc_symbol =</span> SYMBOL) <span class="sc">%&gt;%</span></span>
<span id="cb136-39"><a href="#cb136-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(hgnc_symbol, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb136-40"><a href="#cb136-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">ifelse</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Upregulated"</span>, <span class="st">"Downregulated"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb136-41"><a href="#cb136-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hgnc_symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(heat)) <span class="sc">%&gt;%</span></span>
<span id="cb136-42"><a href="#cb136-42" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(hgnc_symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb136-43"><a href="#cb136-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-44"><a href="#cb136-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation row with labels</span></span>
<span id="cb136-45"><a href="#cb136-45" aria-hidden="true" tabindex="-1"></a>annotation_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Direction =</span> gene_colors<span class="sc">$</span>direction)</span>
<span id="cb136-46"><a href="#cb136-46" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation_row) <span class="ot">&lt;-</span> gene_colors<span class="sc">$</span>hgnc_symbol</span>
<span id="cb136-47"><a href="#cb136-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-48"><a href="#cb136-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a named vector for annotation colors</span></span>
<span id="cb136-49"><a href="#cb136-49" aria-hidden="true" tabindex="-1"></a>annotation_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Direction =</span> <span class="fu">c</span>(<span class="st">"Upregulated"</span> <span class="ot">=</span> <span class="st">"#b2182b"</span>, <span class="st">"Downregulated"</span> <span class="ot">=</span> <span class="st">"#2166ac"</span>))</span>
<span id="cb136-50"><a href="#cb136-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-51"><a href="#cb136-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder columns of heat based on ranked cell types</span></span>
<span id="cb136-52"><a href="#cb136-52" aria-hidden="true" tabindex="-1"></a>heat <span class="ot">&lt;-</span> heat[, ranked_cells_large]</span>
<span id="cb136-53"><a href="#cb136-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-54"><a href="#cb136-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up column names</span></span>
<span id="cb136-55"><a href="#cb136-55" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-positive,"</span>, <span class="st">"+"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-56"><a href="#cb136-56" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"alpha"</span>, <span class="st">"\u03B1"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-57"><a href="#cb136-57" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"beta"</span>, <span class="st">"\u03B2"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-58"><a href="#cb136-58" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gamma"</span>, <span class="st">"\u03B3"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># γ</span></span>
<span id="cb136-59"><a href="#cb136-59" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta"</span>, <span class="st">"\u03B4"</span>, <span class="fu">colnames</span>(heat))  <span class="co"># δ</span></span>
<span id="cb136-60"><a href="#cb136-60" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" t "</span>, <span class="st">" T "</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-61"><a href="#cb136-61" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"nk"</span>, <span class="st">"NK"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-62"><a href="#cb136-62" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">", human"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-63"><a href="#cb136-63" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">" "</span>, <span class="fu">colnames</span>(heat))  <span class="co"># Collapse multiple spaces</span></span>
<span id="cb136-64"><a href="#cb136-64" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">colnames</span>(heat))            <span class="co"># Trim leading/trailing spaces</span></span>
<span id="cb136-65"><a href="#cb136-65" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"of epithelium proper "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-66"><a href="#cb136-66" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"thymus-derived"</span>, <span class="st">"thymus"</span>, <span class="fu">colnames</span>(heat))</span>
<span id="cb136-67"><a href="#cb136-67" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat) <span class="ot">&lt;-</span> <span class="fu">capitalize_first</span>(<span class="fu">colnames</span>(heat))</span>
<span id="cb136-68"><a href="#cb136-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-69"><a href="#cb136-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb136-70"><a href="#cb136-70" aria-hidden="true" tabindex="-1"></a>large_intestine_DEG_SC<span class="ot">&lt;-</span><span class="fu">pheatmap</span>(</span>
<span id="cb136-71"><a href="#cb136-71" aria-hidden="true" tabindex="-1"></a>  heat,                <span class="co"># Filtered expression matrix</span></span>
<span id="cb136-72"><a href="#cb136-72" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,          <span class="co"># Cluster rows (genes) hierarchically</span></span>
<span id="cb136-73"><a href="#cb136-73" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb136-74"><a href="#cb136-74" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cutree_rows = 8, # Cut into 8 clusters</span></span>
<span id="cb136-75"><a href="#cb136-75" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> T,        <span class="co"># Hide row names for cleaner visualization</span></span>
<span id="cb136-76"><a href="#cb136-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">gaps_col =</span> gaps_col_large,</span>
<span id="cb136-77"><a href="#cb136-77" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,                <span class="co"># Normalize rows (genes) for visualization</span></span>
<span id="cb136-78"><a href="#cb136-78" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">rev</span>(<span class="fu">getPalette</span>(<span class="dv">9</span>)),   <span class="co"># Define custom color scheme for the heatmap</span></span>
<span id="cb136-79"><a href="#cb136-79" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontfamily =</span> <span class="st">"Helvetica"</span>,      <span class="co"># Use Helvetica font for better readability</span></span>
<span id="cb136-80"><a href="#cb136-80" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">10</span>,                 <span class="co"># Set font size for axis labels</span></span>
<span id="cb136-81"><a href="#cb136-81" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">12</span>,                <span class="co"># Set font face for axis labels</span></span>
<span id="cb136-82"><a href="#cb136-82" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_row =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb136-83"><a href="#cb136-83" aria-hidden="true" tabindex="-1"></a>  <span class="at">treeheight_col =</span> <span class="dv">0</span>,            <span class="co"># Remove tree height to simplify the plot</span></span>
<span id="cb136-84"><a href="#cb136-84" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fl">13.34</span>,   <span class="co"># in inches</span></span>
<span id="cb136-85"><a href="#cb136-85" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="fl">11.14</span>,   <span class="co"># in inches</span></span>
<span id="cb136-86"><a href="#cb136-86" aria-hidden="true" tabindex="-1"></a>  <span class="at">angle_col =</span> <span class="dv">45</span>, <span class="co"># Rotate column labels</span></span>
<span id="cb136-87"><a href="#cb136-87" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_row =</span> annotation_row,</span>
<span id="cb136-88"><a href="#cb136-88" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> annotation_colors</span>
<span id="cb136-89"><a href="#cb136-89" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb136-90"><a href="#cb136-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-91"><a href="#cb136-91" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the heatmap</span></span>
<span id="cb136-92"><a href="#cb136-92" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(large_intestine_DEG_SC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Data_analysis_to_github_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the heatmap to a PowerPoint file</span></span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph2ppt(x = large_intestine_DEG_SC, "large_intestine_DEG_SC", width = 13.34, height = 11.34)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>